Index: scripts/logwatch.pl
===================================================================
RCS file: /var/cvs/logwatch/scripts/logwatch.pl,v
retrieving revision 1.111
diff -u -r1.111 logwatch.pl
--- scripts/logwatch.pl	21 Jun 2004 15:00:44 -0000	1.111
+++ scripts/logwatch.pl	22 Jun 2004 08:14:35 -0000
@@ -369,7 +369,7 @@
 closedir(LOGFILEDIR);
 
 # Find out what shared functions are defined...
-opendir(SHAREDDIR,$BaseDir . "scripts/shared") or die $BaseDir . "scripts/shared/, no such directory.\n";
+opendir(SHAREDDIR,$BaseDir . "scripts/shared") or die $ConfigDir . "scripts/shared/, no such directory.\n";
 while (defined($ThisFile = readdir(SHAREDDIR))) {
    unless (-d $BaseDir . "scripts/shared/" . $ThisFile) {
       push @AllShared, lc($ThisFile);
