? logwatch-postfix.patch
Index: scripts/services/postfix
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/postfix,v
retrieving revision 1.1
diff -u -r1.1 postfix
--- scripts/services/postfix	3 Nov 2003 04:49:18 -0000	1.1
+++ scripts/services/postfix	17 Nov 2003 08:56:03 -0000
@@ -53,7 +53,6 @@
 	 ( $ThisLine =~ m/^[a-zA-Z0-9]+: client/ ) or 
 	 ( $ThisLine =~ m/^[a-zA-Z0-9]+: message-id/ ) or
 	 ( $ThisLine =~ m/^[a-zA-Z0-9]+: skipped, still being delivered/ ) or
-	 ( $ThisLine =~ m/^[a-zA-Z0-9]+: uid=0 from=<root>/ ) or
 	 ( $ThisLine =~ m/^warning: [\.0-9]+: address not listed for hostname/ ) or
 	 ( $ThisLine =~ m/^[a-zA-Z0-9]+: to\=\<.*>, relay\=.*, delay\=[0-9]+, status\=[sent|deferred]/ ) or
 	 ( $ThisLine =~ m/^warning: [\.0-9]+: hostname .* verification failed: Host not found/ ) ) {
@@ -85,6 +84,11 @@
         $Temp = "From " . $Relay . " to " . $Dest;
         $RelayDenied{$Temp}++;
     }
+    elsif ( ($User,$From) = ($ThisLine =~ /^[a-zA-Z0-9]+: uid=([^ ]*) from=\<([^ ]*)>/)) {
+		#Messages sent by user
+		$Temp = $From . " (uid=" . $User . "): ";
+		$SentBy{$Temp}++;
+	}
 #    elsif ( ($Warning)  = ($ThisLine =~ /^[a-zA-Z0-9]+: Authentication-Warning: [^ ]+: ([^ ]+ set sender to [^ ]+ using -f)/) ) {
 #        $AuthWarns{$Warning}++;
 #    }
@@ -111,7 +115,8 @@
     (keys %ForeignBounce) or
     (keys %LocalBounce) or
     (keys %AuthWarns) or
-    (keys %ForwardErrors)) {
+    (keys %ForwardErrors) or
+	(keys %SentBy)) {
     print "\n\n --------------------- postfix Begin ------------------------ ";
 
     if($MsgsSent > 0) {
@@ -123,12 +128,19 @@
         print "\n\n" . $FourHourReturns . " messages returned after 4 hours";
     }
 
+	if (keys %SentBy) {
+		print "\n\nMessages sent by:\n";
+		foreach $ThisOne (keys %SentBy) {
+			print "    " . $ThisOne . $SentBy{$ThisOne} . " Times(s)\n";
+		}
+	}
+
     if (keys %UnknownUsers) {
-	print "\n\nUnknown users:\n";
-	foreach $ThisOne (keys %UnknownUsers) {
-	    print "    " . $ThisOne . ": " . $UnknownUsers{$ThisOne} . " Times(s)\n";
+		print "\n\nUnknown users:\n";
+		foreach $ThisOne (keys %UnknownUsers) {
+			print "    " . $ThisOne . ": " . $UnknownUsers{$ThisOne} . " Times(s)\n";
+		}
 	}
-    }
 
     if (keys %RelayDenied) {
 	print "\n\nRelaying denied:\n";
@@ -176,3 +188,5 @@
 
 exit(0);
 
+
+# vi: shiftwidth=4 tabstop=4
