--- ./scripts/services/modprobe.org	Sat Oct 12 04:08:18 2002
+++ ./scripts/services/modprobe	Tue Jul  1 09:44:02 2003
@@ -16,10 +16,11 @@
 while (defined($ThisLine = <STDIN>)) {
    chomp($ThisLine);
    if ( $ThisLine =~ s/^modprobe: Can\'t locate module ([\w-]+)\s*$/$1/ ) {
-      $Modules{$ThisLine}++;
-   }
-   else {
-      $OtherList{$ThisLine}++;
+	   $Modules{$ThisLine}++;
+   } elsif ( $ThisLine =~ s/^FATAL: Module (.*) not found./$1/) {
+	   $Modules{$ThisLine}++;
+   } else {
+	   $OtherList{$ThisLine}++;
    }
 }
 
