Index: conf/services/courier.conf
===================================================================
RCS file: /var/cvs/logwatch/conf/services/courier.conf,v
retrieving revision 1.4
diff -u -r1.4 courier.conf
--- conf/services/courier.conf	18 Dec 2003 13:51:27 -0000	1.4
+++ conf/services/courier.conf	7 Mar 2004 21:12:09 -0000
@@ -47,7 +47,7 @@
 # Only give lines pertaining to courier...
 # I'm not sure if this is complete, especially for the new webmail daemon in 0.44.1
 #but you will get at least all currently supported logs 
-*OnlyService = (courierd|courieresmtp|courieresmtpd|courierlocal|imapd|imapd-ssl|pop3d-ssl|pop3d)
+*OnlyService = (courierd|courieresmtp|courieresmtpd|courierlocal|imapd|imapd-ssl)
 
 *RemoveHeaders = 
 
