Index: scripts/services/sendmail-largeboxes
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/sendmail-largeboxes,v
retrieving revision 1.2
diff -u -r1.2 sendmail-largeboxes
--- scripts/services/sendmail-largeboxes	21 Jun 2004 15:07:21 -0000	1.2
+++ scripts/services/sendmail-largeboxes	22 Jun 2004 08:04:22 -0000
@@ -1,10 +1,20 @@
-#!/bin/bash
+#!/bin/sh
 ##########################################################################
 # $Id$
 ##########################################################################
 
-ls -alSh /var/spool/mail | \
-   grep rw | \
-   grep -E "([0-9][0-9]|[4-9])+[0-9]+([.][0-9])*[M]" | \
-   gawk -F ' ' ' { print("Warning: Large mailbox: "$9" ("$5")") } '
- 
+if [ -d /var/mail ]; then
+	SPOOLDIR=/var/mail
+elif [ -d /var/spool/mail ]; then
+	SPOOLDIR=/var/spool/mail
+else
+	echo "Can't find spool directory"
+fi
+
+if [ "$SPOOLDIR" ]; then
+	ls -alSh $SPOOLDIR | \
+		grep rw | \
+		grep -E "([0-9][0-9]|[4-9])+[0-9]+([.][0-9])*[M]" | \
+		gawk -F ' ' ' { print("Warning: Large mailbox: "$9" ("$5")") } '
+fi
+
