Index: scripts/services/samba
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/samba,v
retrieving revision 1.18
diff -u -r1.18 samba
--- scripts/services/samba	3 Feb 2004 02:45:26 -0000	1.18
+++ scripts/services/samba	21 Jun 2004 22:14:36 -0000
@@ -31,38 +31,38 @@
 while (defined($ThisLine = <STDIN>)) {
    chomp($ThisLine);
    if ( 
-      ($ThisLine =~ /Currently not implemented/) or
-      ($ThisLine =~ /version .+ started/) or
-      ($ThisLine =~ /oplock[_ ]break/) or
-      ($ThisLine =~ /No route to host/) or
-      ($ThisLine =~ /response packet id \d+ received with no matching record/) or
-      ($ThisLine =~ /matchname/i) or
-      # Ignore entries in smbmount logfile
-      ($ThisLine =~ /smbmount/) or
-      ($ThisLine =~ /become_local_master/) or
-      ($ThisLine =~ /become_domain_master/) or
       ($ThisLine =~ /add_domain_logon_names/) or
+      ($ThisLine =~ /become_domain_master/) or
+      ($ThisLine =~ /become_local_master/) or
       ($ThisLine =~ /become_logon_server/) or
-      ($ThisLine =~ /start_async_dns/) or
-      ($ThisLine =~ /timeout connecting to/) or
-      ($ThisLine =~ /Operation not permitted/) or
-      ($ThisLine =~ /Record does not exist/) or
+      ($ThisLine =~ /cli_connect\(783\)  Error connecting to [^ ]+ \(Operation already in progress\)$/) or
+      ($ThisLine =~ /closed connection to/) or
       ($ThisLine =~ /Connection reset by peer/) or
-      ($ThisLine =~ /Multiple .+ responses received for a query/) or
       ($ThisLine =~ /Connection timed out/) or
-      ($ThisLine =~ /closed connection to/) or
       ($ThisLine =~ /current master browser/) or
+      ($ThisLine =~ /Currently not implemented/) or
       ($ThisLine =~ /debug_message/) or
-      ($ThisLine =~ /process_name_refresh_request\(184\)  Error - should be sent to WINS server$/) or
-      ($ThisLine =~ /cli_connect\(783\)  Error connecting to [^ ]+ \(Operation already in progress\)$/) or
-      ($ThisLine =~ /nmbd_incomingrequests\.c:process_name_refresh_request\([0-9]+\)$/) or
+      ($ThisLine =~ /get_socket_addr\(\d+\)  getpeername failed. Error was Transport endpoint is not connected$/) or
       ($ThisLine =~ /Got SIGHUP dumping debug info.$/) or
       ($ThisLine =~ /Got SIGTERM: going down/) or
-      ($ThisLine =~ /get_socket_addr\(\d+\)  getpeername failed. Error was Transport endpoint is not connected$/) or
       ($ThisLine =~ /lib\/access.c:check_access\(\d+\)$/) or
-      ($ThisLine =~ /smbd\/process.c:process_smb\(\d+\)$/) or
+      ($ThisLine =~ /matchname/i) or
+      ($ThisLine =~ /Multiple .+ responses received for a query/) or
       ($ThisLine =~ /nmbd\/nmbd_incomingdgrams.c:process_local_master_announce\(\d+\)$/) or
+      ($ThisLine =~ /nmbd_incomingrequests\.c:process_name_refresh_request\([0-9]+\)$/) or
       ($ThisLine =~ /nmbd_namelistdb.c:standard_success_release\(\d+\)  standard_success_release: Name release for name/) or
+      ($ThisLine =~ /No route to host/) or
+      ($ThisLine =~ /Operation not permitted/) or
+      ($ThisLine =~ /oplock[_ ]break/) or
+      ($ThisLine =~ /process_local_message: unknown UDP message command code \(.+\) - ignoring./) or
+      ($ThisLine =~ /process_name_refresh_request\(184\)  Error - should be sent to WINS server$/) or
+      ($ThisLine =~ /Record does not exist/) or
+      ($ThisLine =~ /response packet id \d+ received with no matching record/) or
+      ($ThisLine =~ /smbd\/process.c:process_smb\(\d+\)$/) or
+      ($ThisLine =~ /smbmount/) or
+      ($ThisLine =~ /start_async_dns/) or
+      ($ThisLine =~ /timeout connecting to/) or
+      ($ThisLine =~ /version .+ started/) or
 		($ThisLine =~ /===============================================================/)
    ) {
       #Don't care about these...
