Index: conf/services/pam_unix.conf
===================================================================
RCS file: /var/cvs/logwatch/conf/services/pam_unix.conf,v
retrieving revision 1.2
diff -u -r1.2 pam_unix.conf
--- conf/services/pam_unix.conf	12 Oct 2002 02:08:09 -0000	1.2
+++ conf/services/pam_unix.conf	2 Oct 2004 11:06:03 -0000
@@ -16,6 +16,7 @@
 
 # Which logfile group...
 LogFile = messages
+LogFile = secure
 
 ########################################################
 # This was written and is maintained by:
