Index: scripts/services/http
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/http,v
retrieving revision 1.14
diff -u -r1.14 http
--- scripts/services/http	29 Oct 2004 21:19:00 -0000	1.14
+++ scripts/services/http	31 Oct 2004 15:19:03 -0000
@@ -119,7 +119,7 @@
 my $host =           "";
 
 ######################
-my $image_types =    '(\.bmp|\.cdr|\.emz|\.gif|\.ico|\.jpeg|\.jpg|\.png|\.sxd|\.tif|\.tiff|\.wbmp|\.wmf|\.wmz|\.xdm)';
+my $image_types =    '(\.bmp|\.cdr|\.emz|\.gif|\.ico|\.jpeg|\.jpg|\.png|\.svg|\.sxd|\.tif|\.tiff|\.wbmp|\.wmf|\.wmz|\.xdm)';
 my $content_types =  '(';
    $content_types =  $content_types.'\/server-status|\/server-info';
    $content_types =  $content_types.'|\.htm|\.html|\.jhtml|\.phtml|\.shtml|\/';
