Index: scripts/services/samba
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/samba,v
retrieving revision 1.12
diff -u -r1.12 samba
--- scripts/services/samba	26 Nov 2003 14:36:30 -0000	1.12
+++ scripts/services/samba	11 Dec 2003 10:27:24 -0000
@@ -48,6 +48,8 @@
 		($ThisLine =~ /Got SIGHUP/) or
 		($ThisLine =~ /current master browser/) or
 		($ThisLine =~ /debug_message/) or
+		($ThisLine =~ /process_name_refresh_request\(184\)  Error - should be sent to WINS server$/)or
+		($ThisLine =~ /cli_connect\(783\)  Error connecting to [^ ]+ \(Operation already in progress\)$/)or
 		($ThisLine =~ /===============================================================/)
 	) {
 		#Don't care about these...
@@ -57,8 +59,15 @@
 		$NoServ{$NoService}++;
 	} elsif ($ThisLine =~ s/Denied connection from\s+(\S+)$/$1/) {
 		$Denied{$ThisLine}++;
-	} elsif ( ($Ip,$Name,$Group,$Subnet) = ($ThisLine =~ /Denied connection from  \(([^ ]+)\)  Samba name server ([^ ]+) is now a local master browser for workgroup ([^ ]+) on subnet ([^ ]+)/ ) ) {
-		$Denied{$Ip}++;
+	} elsif ($ThisLine =~ s/Connection denied from\s+(\S+)$/$1/) {
+		$Denied{$ThisLine}++;
+	} elsif ( ($Where,$Ip,$Browser) = ($ThisLine =~ /(.*)  Denied connection from  \(([^ ]+)\)  Doing a node status request to the domain master browser at IP ([^ ]+) failed.  Cannot get workgroup name./ ) ) {
+		$Temp = "$Where  ($Ip)";
+		$Denied{$Temp}++;
+		$CantGetGroup{$Browser}++;
+	} elsif ( ($Where,$Ip,$Name,$Group,$Subnet) = ($ThisLine =~ /(.*)  Denied connection from  \(([^ ]+)\)  Samba name server ([^ ]+) is now a local master browser for workgroup ([^ ]+) on subnet ([^ ]+)/ ) ) {
+		$Temp = "$Where  ($Ip)";
+		$Denied{$Temp}++;
 		$BeLocalMaster{$Subnet}{$Group}{$Name}++;
 	} elsif (($User) = $ThisLine =~ /rejected invalid user ([^ ]+)/ ) {
 		$InvalidUser{$User}++;
@@ -74,8 +83,8 @@
 	} elsif ( ($Server,$Ip,$Group) = ($ThisLine =~ /Server ([^ ]+) at IP ([^ ]+) is announcing itself as a local master browser for workgroup ([^ ]+) and we think we are master. Forcing election.$/ ) ) {
 		$Temp = $Server . "(" . $Ip . ")";
 		$ForceElection{$Group}{$Temp}++;
-	} elsif ( ($Server,$Ip,undef) = ($ThisLine =~ /process_name_refresh_request: unicast name registration request received for name ([^ ]+) from IP ([^ ]+) on subnet ([^ ]+)./ ) ) {
-		$Temp = "$Server ($Ip)";
+	} elsif ( (undef,$Command,$Server,$Ip,undef) = ($ThisLine =~ /([^ ]+): unicast name ([^ ]+) request received for name ([^ ]+) from IP ([^ ]+) on subnet (.*)\./ ) ) {
+		$Temp = "$Command on subnet $Subnet : $Server ($Ip)";
 		$UnicastRegister{$Temp}++;
 	} elsif ( ($Group,$Subnet) = ($ThisLine =~ /standard_fail_register: Failed to register\/refresh name ([^ ]+) on subnet ([^ ]+)$/ ) ) {
 		$FailedRegister{$Subnet}{$Group}++;
@@ -88,16 +97,27 @@
 		$Crash{$Temp}++;
 	} elsif ( ( $ThisLine =~ /get_domain_master_name_node_status_fail\(([^ ]+)\)/ ) ) {
 		$GetDomainMasterStatusFail++;
+	} elsif ( ($User) = ($ThisLine =~ /pass_check_smb\(552\)  Account for user '([^ ]+)' was disabled.$/) ) {
+		$AccountDisabled{$User}++;
+	} elsif ( ($Version) = ($ThisLine =~ /Discarding invalid wins\.dat file \[(.*)\]$/) ) {
+		$DiscardWins{$Version}++;
 	} else {
 		# Report any unmatched entries...
 		$OtherList{$ThisLine}++;
+		#
+		#TODO:
+		#lib/util_sock.c:get_socket_addr(1012)  getpeername failed. Error was Transport endpoint is not connected
+		#lib/util_sock.c:send_smb(704)  Error writing 4 bytes to client. -1. (Broken pipe)
+		#lib/util_sock.c:write_socket(524)  write_socket: Error writing 4 bytes to socket 5: ERRNO = Broken pipe
+		#lib/util_sock.c:write_socket_data(499)  write_socket_data: write failure. Error = Broken pipe
+		#lib/util_sock.c:read_data(436)  read_data: read failure for 4. Error = Brak drogi do systemu
 	}
 }
 
 #########################################
 #
 
-if (%Crash) {
+if (keys %Crash) {
 	print "\nWARNING!!!!!!\n";
 	print "Server crashed:\n";
 	foreach $Dead (sort {$a cmp $b} keys %Crash) {
@@ -105,6 +125,13 @@
 	}
 }
 
+if (keys %DiscardWins) {
+	print "\nDiscarded invalid wins.dat file with version:\n";
+	foreach $Version (sort {$a cmp $b} keys %DiscardWins) {
+		print "   $Version : $DiscardWins{$Version} Time(s)\n";
+	}
+}
+
 if (($Detail >= 5) and (keys %Connect)) {
    print "\nOpened Sessions:\n";
    foreach $Serv (sort {$a cmp $b} keys %Connect) {
@@ -120,7 +147,7 @@
 
 if (keys %Denied) {
    print "\nConnections Denied:\n";
-   foreach $Line (keys %Denied) {
+   foreach $Line (sort {$a cmp $b} keys %Denied) {
       print "   $Line : $Denied{$Line} Time(s)\n";
    }
 }
@@ -167,7 +194,7 @@
 }
 
 if (($Detail >= 5) and (keys %UnicastRegister)) {
-	print "\nUnicast name registration requests:\n";
+	print "\nUnicast name requests:\n";
 	foreach $ThisOne (sort {$a cmp $b} keys %UnicastRegister) {
 		print "   $ThisOne : $UnicastRegister{$ThisOne} Time(s)\n";
 	}
@@ -199,23 +226,30 @@
 
 if (keys %InvalidUser) {
    print "\nInvalid Users:\n";
-   foreach $Line (keys %InvalidUser) {
+   foreach $Line (sort {$a cmp $b} keys %InvalidUser) {
       print "   $Line : $InvalidUser{$Line} Time(s)\n";
    }
 }
 
 if (keys %NotFoundUser) {
    print "\nUsers not found in UNIX Database:\n";
-   foreach $Line (keys %NotFoundUser) {
+   foreach $Line (sort {$a cmp $b} keys %NotFoundUser) {
       print "   $Line : $NotFoundUser{$Line} Time(s)\n";
    }
 }
 
 if (keys %RejectedUser) {
    print "\nRejected Users:\n";
-   foreach $Line (keys %RejectedUser) {
+   foreach $Line (sort {$a cmp $b} keys %RejectedUser) {
       print "   $Line : $RejectedUser{$Line} Time(s)\n";
    }
+}
+
+if (keys %AccountDisabled) {
+	print "\nAccounts disabled:\n";
+	foreach $User (sort {$a cmp $b} keys %AccountDisabled) {
+		print "   $User : $AccountDisabled{$User} Time(s)\n";
+	}
 }
 
 if ($SocketReadError > 0) {
