Index: scripts/services/clam-update
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/clam-update,v
retrieving revision 1.6
diff -u -r1.6 clam-update
--- scripts/services/clam-update	21 Jun 2004 14:59:05 -0000	1.6
+++ scripts/services/clam-update	21 Jun 2004 18:00:50 -0000
@@ -216,10 +216,10 @@
 }
 
 if ($NotificationOK > 0) {
-   print "Clamd successfully notified about the update $NotificationOK Time(s).\n";
+   print "  Clamd successfully notified about the update $NotificationOK Time(s).\n";
 } elsif (($MainUpdated or $DailyUpdated) and ($NotificationOK > 0)) {
-   print "WARNING\n";
-   print "Databases are updated, but Clamd is not notified.\n";
+   print "  WARNING\n";
+   print "  Databases are updated, but Clamd is not notified.\n";
 }
 
 if (($Updated) and ($Detail >= 10)) {
