Index: scripts/services/http
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/http,v
retrieving revision 1.5
diff -u -r1.5 http
--- scripts/services/http	12 Nov 2003 04:09:08 -0000	1.5
+++ scripts/services/http	9 Dec 2003 12:55:29 -0000
@@ -39,6 +39,16 @@
 	my $a1xx_resp = 0;
 	my $image_count = 0;
 	my $image_bytes = 0;
+	my $docs_count = 0;
+	my $docs_bytes = 0;
+	my $archive_count = 0;
+	my $archive_bytes = 0;
+	my $sound_count = 0;
+	my $sound_bytes = 0;
+	my $movie_count = 0;
+	my $movie_bytes = 0;
+	my $winexec_count = 0;
+	my $winexec_bytes = 0;
 	my $content_count = 0;
 	my $content_bytes = 0;
 	my $redirect_count = 0;
@@ -46,12 +56,18 @@
 	my $other_count = 0;
 	my $other_bytes = 0;
 	my $total_hack_count = 0;
-	my $image_types = '(\.jpg|\.gif|\.png|\.ico|\.swf|\.wav)';
+	my $image_types = '(\.bmp|\.gif|\.ico|\.jpg|\.png|\.swf)';
 	my $content_types = '(';
 		$content_types = $content_types.'\/server-status|\/server-info';
-		$content_types = $content_types.'|\.css|\.htm|\.html|\.txt|\/';
-		$content_types = $content_types.'|\.shtml|\.php|\.asp|\.pl|\/';
-		$content_types = $content_types.'|\.js|\.cgi)';
+		$content_types = $content_types.'|\.htm|\.html|\.phtml|\.shtml|\/';
+		$content_types = $content_types.'|\.php|\.php3|\.asp|\.pl|\/';
+		$content_types = $content_types.'|\.css|\.js|\.cgi|\.txt|\/';
+		$content_types = $content_types.'|\.class|\.jsp|\.jar)';
+	my $docs_types = '(\.doc|\.pdf|\.rtf|\.xls|\.tex)';
+	my $archive_types = '(\.zip|\.bz2|\.gz)';
+	my $sound_types = '(\.mp3|\.wav|\.wma)';
+	my $movie_types = '(\.avi)';
+	my $winexec_types = '(\.exe)';
 
 #
 #   what to look for as an attack  USE LOWER CASE!!!!!!
@@ -152,26 +168,38 @@
 	if ($field{lc_url} =~ /$image_types$/) {
 		$image_count += 1;
 		$image_bytes += $field{bytes_transfered};
-	}
-		elsif ($field{base_url} =~ /$content_types$/) {
-			$content_count += 1;
-			$content_bytes += $field{bytes_transfered};
-		}
-			elsif ($field{http_rc} =~ /3\d\d/) {
-				$redirect_count += 1;
-				$redirect_bytes += $field{bytes_transfered};
+	} elsif ($field{lc_url} =~ /$docs_types$/) {
+		$docs_count += 1;
+		$docs_bytes += $field{bytes_transfered};
+	} elsif ($field{lc_url} =~ /$archive_types$/) {
+		$archive_count += 1;
+		$archive_bytes += $field{bytes_transfered};
+	} elsif ($field{lc_url} =~ /$sound_types$/) {
+		$sound_count += 1;
+		$sound_bytes += $field{bytes_transfered};
+	} elsif ($field{lc_url} =~ /$movie_types$/) {
+		$movie_count += 1;
+		$movie_bytes += $field{bytes_transfered};
+	} elsif ($field{lc_url} =~ /$winexec_types$/) {
+		$winexec_count += 1;
+		$winexec_bytes += $field{bytes_transfered};
+	} elsif ($field{base_url} =~ /$content_types$/) {
+		$content_count += 1;
+		$content_bytes += $field{bytes_transfered};
+	} elsif ($field{http_rc} =~ /3\d\d/) {
+		$redirect_count += 1;
+		$redirect_bytes += $field{bytes_transfered};
+	} else {
+		$other_count += 1;
+		$other_bytes += $field{bytes_transfered};
+		if (!$isahack ) {
+			if ( !$ignore_error_hacks ) {
+				$needs_exam{$field{request}} .= $field{http_rc}." ";
+			} elsif ( $field{http_rc} < 400 ) {
+				$needs_exam{$field{request}} .= $field{http_rc}." ";
 			}
-				else {
-					$other_count += 1;
-					$other_bytes += $field{bytes_transfered};
-					if (!$isahack ) {
-						if ( !$ignore_error_hacks ) {
-							$needs_exam{$field{request}} .= $field{http_rc}." ";
-						} elsif ( $field{http_rc} < 400 ) {
-							$needs_exam{$field{request}} .= $field{http_rc}." ";
-						}
-					}
-				}
+		}
+	}
 ##
 ## Do the > 4 stuff
 ## 
@@ -202,6 +230,11 @@
 		print  " (1xx $a1xx_resp, 2xx $a2xx_resp, 3xx $a3xx_resp,";
 		print  " 4xx $a4xx_resp, 5xx $a5xx_resp) \n";
 		print	" ".$image_count." Images (".$image_bytes." bytes),\n"; 
+		print   " ".$docs_count." Documents (".$docs_bytes." bytes),\n";
+		print   " ".$archive_count." Archives (".$archive_bytes." bytes),\n";
+		print   " ".$sound_count." Sound files (".$sound_bytes." bytes),\n";
+		print   " ".$movie_count." Movies files (".$movie_bytes." bytes),\n";
+		print   " ".$winexec_count." Windows executable files (".$winexec_bytes." bytes),\n";
 		print	" ".$content_count." Content pages (".$content_bytes." bytes),\n"; 
 		print	" ".$redirect_count." Redirects (".$redirect_bytes." bytes),\n"; 
 		print	" ".$other_count." Other (".$other_bytes." bytes) \n";
@@ -270,3 +303,5 @@
 	}
 
 exit (0);
+
+# vi: shiftwidth=4 tabstop=4
