--- logwatch-7.3.6/syslog-ng	2008-02-02 12:52:34.000000000 +0200
+++ logwatch-7.3.6-syslog/syslog-ng	2008-10-06 09:51:38.269584080 +0300
@@ -59,15 +59,24 @@
    }
 
    #TD syslog-ng[9754]: syslog-ng version 1.6.2 starting
-   elsif ($ThisLine =~ /^syslog-ng version ([0-9]*\.)*[0-9]* starting$/) {
+   #TD syslog-ng[3956]: syslog-ng starting up; version='2.0.9'
+   elsif ($ThisLine =~ /^syslog-ng version [\d.]+ starting$/ ||
+      $ThisLine =~ /^syslog-ng starting up; version='[\d.]+'$/) {
       $Starts++;
    }
 
    #TD syslog-ng[9754]: syslog-ng version 1.6.2 going down
-   elsif ($ThisLine =~ /^syslog-ng version ([0-9]*\.)*[0-9]* going down$/) {
+   #TD syslog-ng[20043]: syslog-ng shutting down; version='2.0.9'
+   elsif ($ThisLine =~ /^syslog-ng version [\d.]+ going down$/ ||
+      $ThisLine =~ /^syslog-ng shutting down; version='[\d.]+'$/) {
       $Stops++;
    }
 
+   #TD syslog-ng[20043]: Termination requested via signal, terminating;
+   elsif ($ThisLine =~ /^Termination requested via signal, terminating;/) {
+      # happens with shutdown, but it's not for extra accounting
+   }
+
    #TD syslog-ng[4833]: STATS: dropped 0
    elsif ($ThisLine =~ /^STATS: dropped ([0-9]*)$/) {
       if ($1 != 0) { $Drops = $Drops + $1; }
--- logwatch-7.3.6/syslog-ng	2008-10-27 10:15:07.818460828 +0200
+++ logwatch-7.3.6/syslog-ng	2008-10-27 10:18:37.748827803 +0200
@@ -58,6 +58,11 @@
       } else { $Starts++; }
    }
 
+   #TD syslog-ng[4027]: Configuration reload request received, reloading configuration;
+   elsif ($ThisLine =~ /^Configuration reload request received, reloading configuration;/) {
+      $Reloads++;
+   }
+
    #TD syslog-ng[9754]: syslog-ng version 1.6.2 starting
    #TD syslog-ng[3956]: syslog-ng starting up; version='2.0.9'
    elsif ($ThisLine =~ /^syslog-ng version [\d.]+ starting$/ ||
