--- ./conf/services/pureftpd.conf.org	Mon Nov  3 15:25:20 2003
+++ ./conf/services/pureftpd.conf	Fri Nov  7 10:41:44 2003
@@ -15,7 +15,7 @@
 Title = "pureftp"
 
 # Which logfile group...
-LogFile = pureftp
+LogFile = messages
 
 *OnlyService = pure-ftpd
 *RemoveHeaders =
@@ -30,5 +30,5 @@
 
 # How many connections did we get?
 # This only shows where it came from and how many per ip / host.
-$show_new_connections = 0
+$show_new_connections = 1
 
 # vi: shiftwidth=3 tabstop=3 et
