Index: scripts/services/http
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/http,v
retrieving revision 1.9
diff -u -r1.9 http
--- scripts/services/http	3 Feb 2004 02:45:26 -0000	1.9
+++ scripts/services/http	31 Mar 2004 13:15:20 -0000
@@ -79,39 +79,45 @@
 my $wpad_bytes =     0;
 my $src_count =      0;
 my $src_bytes =      0;
+my $logs_count =     0;
+my $logs_bytes =     0;
 my $images_count =   0;
 my $images_bytes =   0;
-my $image_types =    '(\.bmp|\.emz|\.gif|\.ico|\.jpeg|\.jpg|\.png|\.swf|\.tif|\.tiff|\.wmf|\.wmz|\.xdm)';
+my $image_types =    '(\.bmp|\.emz|\.gif|\.ico|\.jpeg|\.jpg|\.png|\.swf|\.sxd|\.tif|\.tiff|\.wmf|\.wmz|\.xdm)';
 my $content_types =  '(';
    $content_types =  $content_types.'\/server-status|\/server-info';
    $content_types =  $content_types.'|\.htm|\.html|\.jhtml|\.phtml|\.shtml|\/';
    $content_types =  $content_types.'|\.php|\.php3|\.asp|\.pl|\.wml|\/';
    $content_types =  $content_types.'|\.css|\.js|\.cgi|\.txt|\/';
    $content_types =  $content_types.'|\.class|\.jsp|\.jar|\.java)';
-my $docs_types =     '(\.doc|\.dvi|\.gnumeric|\.mcd|\.mso|\.pdf|\.pps|\.ppt|\.ps|\.rtf|\.tex|\.text|\.xls|\.xml)';
-my $archive_types =  '(\.ace|\.bz2|\.cab|\.deb|\.gz|\.hqx|\.rar|\.rpm|\.tbz2|\.zip)';
+my $docs_types =     '(\.asc|\.doc|\.dot|\.dvi|\.gnumeric|\.mcd|\.mso|\.pdf|\.pps|\.ppt|\.ps|\.rtf|\.tex|\.text|\.xls|\.xml)';
+my $archive_types =  '(\.ace|\.bz2|\.cab|\.deb|\.ed2k|\.gz|\.hqx|\.rar|\.rpm|\.sign|\.tbz2|\.tgz|\.Z|\.zip)';
 my $sound_types =    '(\.mp3|\.ram|\.raw|\.rm|\.wav|\.wma|\.mid)';
-my $movie_types =    '(\.asf|\.avi|\.mid|\.mpg|\.mpeg|\.mov)';
+my $movie_types =    '(\.asf|\.ass|\.avi|\.idx|\.mid|\.mpg|\.mpeg|\.mov|\.qt|\.sub)';
 my $winexec_types =  '(\.com|\.exe|\.dll)';
 my $wpad_files =     '(wpad\.dat|wspad\.dat|proxy\.pac)';
 my $program_src =    '(\.bas|\.c|\.diff|\.f|\.h|\.patch|\.cpp)';
-my $images_types =   '(\.bin|\.cue|\.iso|\.run)';
+my $images_types =   '(\.bin|\.cue|\.img|\.iso|\.run)';
+my $logs_types =     '(\.log|\.logs)';
 
 #
 #   what to look for as an attack  USE LOWER CASE!!!!!!
 #
 my @exploits = (
-   '\\x90\\x02\\xb1\\x02\\xb1',
-   '\\x02\\xb1\\x02\\xb1',
-   '\\x90\\x90\\x90\\x90',
-   '\\x04\\x01',
-   '\\x05\\x01',
+   '\\\x90\\\x02\\\xb1\\\x02\\\xb1',
+   '\\\x02\\\xb1',
+   '\\\x90\\\x90\\\x90\\\x90',
+   '\\\x04\\\x01',
+   '\\\x05\\\x01',
    '\/c\+dir',
    'cmd.exe',
+   'cltreq.asp',
    'default.ida',
    'nsiislog.dll',
+   'owssvr.dll',
    'phpmyadmin',
    'root.exe',
+   'shtml.exe',
    'win.ini'
 );
 
@@ -238,6 +244,9 @@
    } elsif ($field{base_url} =~ /$images_types$/) {
       $images_count += 1;
       $images_bytes += $field{bytes_transfered};
+   } elsif ($field{base_url} =~ /$logs_types$/) {
+      $logs_count += 1;
+      $logs_bytes += $field{bytes_transfered};
    } elsif ($field{http_rc} =~ /3\d\d/) {
       $redirect_count += 1;
       $redirect_bytes += $field{bytes_transfered};
@@ -293,18 +302,19 @@
    print " responses ";
    print " (1xx $a1xx_resp, 2xx $a2xx_resp, 3xx $a3xx_resp,";
    print " 4xx $a4xx_resp, 5xx $a5xx_resp) \n";
-   print	  " $image_count Images ($image_bytes bytes),\n";
-   print   " $docs_count Documents ($docs_bytes bytes),\n";
-   print   " $archive_count Archives ($archive_bytes bytes),\n";
-   print   " $sound_count Sound files ($sound_bytes bytes),\n";
-   print   " $movie_count Movies files ($movie_bytes bytes),\n";
-   print   " $winexec_count Windows executable files ($winexec_bytes bytes),\n";
-   print	  " $content_count Content pages ($content_bytes bytes),\n";
-   print	  " $redirect_count Redirects ($redirect_bytes bytes),\n";
-   print   " $wpad_count Proxy Configuration Files ($wpad_bytes bytes),\n";
-   print   " $src_count Program source files ($src_bytes bytes),\n";
-   print   " $images_count CD Images ($images_bytes bytes),\n"; 
-   print	  " $other_count Other ($other_bytes bytes) \n";
+   if ($image_count > 0)      { print " $image_count Images ($image_bytes bytes),\n"; }
+   if ($docs_count > 0)       { print " $docs_count Documents ($docs_bytes bytes),\n"; }
+   if ($archive_count > 0)    { print " $archive_count Archives ($archive_bytes bytes),\n"; }
+   if ($sound_count > 0)      { print " $sound_count Sound files ($sound_bytes bytes),\n"; }
+   if ($movie_count > 0)      { print " $movie_count Movies files ($movie_bytes bytes),\n"; }
+   if ($winexec_count > 0)    { print " $winexec_count Windows executable files ($winexec_bytes bytes),\n"; }
+   if ($content_count > 0)    { print " $content_count Content pages ($content_bytes bytes),\n"; }
+   if ($redirect_count > 0)   { print " $redirect_count Redirects ($redirect_bytes bytes),\n"; }
+   if ($wpad_count > 0)       { print " $wpad_count Proxy Configuration Files ($wpad_bytes bytes),\n"; }
+   if ($src_count > 0)        { print " $src_count Program source files ($src_bytes bytes),\n"; }
+   if ($images_count > 0)     { print " $images_count CD Images ($images_bytes bytes),\n"; }
+   if ($logs_count > 0)       { print " $logs_count various Logs ($logs_bytes bytes),\n"; }
+   if ($other_count > 0)      { print " $other_count Other ($other_bytes bytes) \n"; }
 }
 
 #
