Index: scripts/services/postfix
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/postfix,v
retrieving revision 1.12
diff -u -r1.12 postfix
--- scripts/services/postfix	21 Jun 2004 14:59:05 -0000	1.12
+++ scripts/services/postfix	21 Jun 2004 21:45:27 -0000
@@ -105,32 +105,36 @@
 
 while (defined($ThisLine = <STDIN>)) {
    if ( 
-      ( $ThisLine =~ m/^connect/ ) or 
-      ( $ThisLine =~ m/^disconnect/ ) or 
-      ( $ThisLine =~ m/^reload configuration/ ) or 
       ( $ThisLine =~ m/^[a-zA-Z0-9]+: client=([^ ]*\[[^ ]*\])$/ ) or 
       ( $ThisLine =~ m/^[a-zA-Z0-9]+: message-id/ ) or
       ( $ThisLine =~ m/^[a-zA-Z0-9]+: skipped, still being delivered/ ) or
-      ( $ThisLine =~ m/^warning: [\.0-9]+: address not listed for hostname/ ) or
       ( $ThisLine =~ m/^[a-zA-Z0-9]+: to\=\<.*>, relay\=.*, delay\=[0-9]+, status\=(sent|deferred)/ ) or
-      ( $ThisLine =~ m/^warning: [\.0-9]+: hostname .* verification failed: Host not found/ ) or
-      ( $ThisLine =~ m/^warning: no MX host for .* has a valid A record$/ ) or
-      ( $ThisLine =~ m/^warning: numeric domain name in resource data of MX record for .*$/ ) or
+      ( $ThisLine =~ m/^Deleted: \d message$/ ) or
+      ( $ThisLine =~ m/^Peer certificate could not be verified$/ ) or
+      ( $ThisLine =~ m/^Peer certficate could not be verified$/ ) or #postfix typo
+      ( $ThisLine =~ m/^Peer verification:/ ) or
+      ( $ThisLine =~ m/^SSL_accept error from/ ) or
+      ( $ThisLine =~ m/^connect/ ) or 
       ( $ThisLine =~ m/^daemon started$/ ) or
+      ( $ThisLine =~ m/^daemon started -- version / ) or
+      ( $ThisLine =~ m/^dict_eval_action:/ ) or
+      ( $ThisLine =~ m/^disconnect/ ) or 
+      ( $ThisLine =~ m/^mynetworks:/ ) or
+      ( $ThisLine =~ m/^name_mask:/ ) or
+      ( $ThisLine =~ m/^reload configuration/ ) or 
+      ( $ThisLine =~ m/^setting up TLS connection (from|to)/ ) or
       ( $ThisLine =~ m/^terminating on signal 15$/ ) or
-      ( $ThisLine =~ m/^warning: Mail system is down -- accessing queue directly$/ ) or
-      ( $ThisLine =~ m/^Deleted: \d message$/ ) or
       ( $ThisLine =~ m/^warning: [a-zA-Z0-9]+: skipping further client input$/ ) or
+      ( $ThisLine =~ m/^warning: [\.0-9]+: address not listed for hostname/ ) or
+      ( $ThisLine =~ m/^warning: [\.0-9]+: hostname .* verification failed: Host not found/ ) or
+      ( $ThisLine =~ m/^warning: Mail system is down -- accessing queue directly$/ ) or
+      ( $ThisLine =~ m/^warning: SASL authentication failure: Password verification failed$/ ) or
+      ( $ThisLine =~ m/^warning: SASL authentication failure: no secret in database$/ ) or
+      ( $ThisLine =~ m/^warning: no MX host for .* has a valid A record$/ ) or
+      ( $ThisLine =~ m/^warning: numeric domain name in resource data of MX record for .*$/ ) or
       ( $ThisLine =~ m/^warning: premature end-of-input from cleanup socket while reading input attribute name$/ ) or
       ( $ThisLine =~ m/^warning: uid=\d: Broken pipe$/ ) or
-      ( $ThisLine =~ m/^warning: SASL authentication failure: no secret in database$/ ) or
-      ( $ThisLine =~ m/^warning: SASL authentication failure: Password verification failed$/ ) or
-      ( $ThisLine =~ m/^setting up TLS connection (from|to)/ ) or
-      ( $ThisLine =~ m/^SSL_accept error from/ ) or
-      ( $ThisLine =~ m/^verify error:num=/ ) or
-      ( $ThisLine =~ m/^Peer verification:/ ) or
-      ( $ThisLine =~ m/^Peer certificate could not be verified$/ ) or
-      ( $ThisLine =~ m/^Peer certficate could not be verified$/ ) #postfix typo
+      ( $ThisLine =~ m/^verify error:num=/ )
    ) {
       # We don't care about these
    } elsif ( ($Bytes) = ($ThisLine =~ /^[a-zA-Z0-9]+: from=.*size=([0-9]+).*$/) ) {
