--- ./scripts/services/secure.org	Tue Feb  3 05:26:36 2004
+++ ./scripts/services/secure	Thu Feb 19 10:57:06 2004
@@ -47,6 +47,7 @@
       ( $ThisLine =~ /^sudo:/) or
       ( $ThisLine =~ /^halt:/) or
       ( $ThisLine =~ /^reboot:/) or
+      ( $ThisLine =~ /^pam_xauth\[\d+\]: call_xauth: child returned \d/) or
       ( $ThisLine =~ /^passwd\[\d+\]:/) or
       ( $ThisLine =~ /warning: can.t get client address: Connection refused/) or
       ( $ThisLine =~ /^xinetd\[\d+\]: USERID: ([^ ]+) (.+)$/ ) or
@@ -108,7 +109,7 @@
       $Error{$Service}{$Err}++;
    } elsif ( ($Service,$Err) = ($ThisLine =~ /^([^ ]+): (password mismatch for [^ ]+ in [^ ]+):.*$/ ) ) {
       $Error{$Service}{$Err}++;
-   } elsif ( $ThisLine =~ /^login: ROOT LOGIN/) {
+   } elsif ( $ThisLine =~ /^login: ROOT LOGIN ON tty[0-9]+/) {
       $RootLoginTTY++
    } elsif ( (undef,$User) = ($ThisLine =~ /^login: LOGIN ON (tty|pts\/)[0-9]+ BY ([^ ]+)/ )) {
       $UserLogin{$User}++;
@@ -122,7 +123,7 @@
       push @DeletedGroups, "   $ThisLine\n";
    } elsif ( $ThisLine =~ s/^(useradd|adduser)\[\d+\]: new group: name=(.+), gid=(\d+).*$/$1 ($2)/ ) {
       push @NewGroups, "   $ThisLine\n";
-   } elsif ( ($User,$Group) =~ /useradd\[\d+\]: add `([^ ]+)' to group `([^ ]+)'$/ ) {
+   } elsif ( (undef,$User,$Group) = ($ThisLine =~ /(usermod|useradd)\[\d+\]: add `([^ ]+)' to group `([^ ]+)'/ )) {
       $AddToGroup{$Group}{$User}++;
    } elsif ( $ThisLine =~ s/^groupadd\[\d+\]: new group: name=(.+), gid=(\d+).*$/$1 ($2)/ ) {
       push @NewGroups, "   $ThisLine\n";
@@ -141,8 +142,10 @@
       $Refused->{$service}->{$from}++;
    } elsif ( ($User) = ($ThisLine =~ /^chage\[\d+\]: changed password expiry for ([^ ]+)/)) {
       $PasswordExpiry{$User}++;
-   } elsif ( (undef) = ($ThisLine =~ /^pam_console\[\d+\]: console file lock already in place ([^ ]+)/)) {
+   } elsif ( (undef) = ($ThisLine =~ /^pam_console\[\d+\]: console file lock already in place ([^ ]+)/ )) {
       $ConsoleLock++;
+   } elsif ( ($Message) = ($ThisLine =~ /^pam_xauth\[\d+\]: call_xauth: (.+)/)) {
+      $XauthMessage{$Message}++;
    } else {
       # Unmatched entries...
       push @OtherList, "$ThisLine\n";
@@ -170,8 +173,8 @@
 if (keys %AddToGroup) {
    print "\nAdded User to group:\n";
    foreach $Group (sort {$a cmp $b} keys %AddToGroup) {
-      print "   Group $Group:\n";
-      foreach $User (sort {$a cmp $b} keys %{$AddToGroup{$User}}) {
+      print "   $Group:\n";
+      foreach $User (sort {$a cmp $b} keys %{$AddToGroup{$Group}}) {
          print "      $User\n";
       }
    }
@@ -264,6 +267,13 @@
    }
 }
 
+if (keys %XauthMessage) {
+   print "\nReported by call_xauth:\n";
+   foreach $Message (sort {$a cmp $b} keys %XauthMessage) {
+      print "   $Message : $XauthMessage{$Message} Time(s)\n";
+   }
+}
+
 if ($#SudoList >= 0) {
    print "\nUnauthorized sudo commands attempted (" . ($#SudoList + 1) . "):\n";
    print @SudoList;
