Index: scripts/services/sshd
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/sshd,v
retrieving revision 1.33
diff -u -r1.33 sshd
--- scripts/services/sshd	15 Dec 2003 19:01:19 -0000	1.33
+++ scripts/services/sshd	23 Dec 2003 22:41:08 -0000
@@ -2,6 +2,8 @@
 ##########################################################################
 # $Id$
 ##########################################################################
+# $Log$
+##########################################################################
 
 ########################################################
 # This was written and is maintained by:
@@ -71,8 +73,11 @@
       }
       $BadLogins{"$2/$1 from $3"}++;
    } elsif ( $ThisLine =~ m/^Failed (\S+) for illegal user (\S+) from ([^ ]+) port (\d+)/ ) { #openssh
-      $BadLogins{"$2/$1 from $3"}++;
-      $IllegalUsers{$2}++;
+      $Temp = "$2/$1 from $3";
+      $BadLogins{$Temp}++;
+      $IllegalUsers{$Temp}++;
+   } elsif ( ($User) = ( $ThisLine =~ /Disconnecting: Too many authentication failures for ([^ ]+)/)) {
+      $TooManyFailures{$User}++;
    } elsif ( ($User) = ($ThisLine =~ /^input_userauth_request: illegal user ([^ ]+)$/ )) {
       $IllegalUsers{"$User/none from unknown"}++;
    } elsif ( $ThisLine =~ m/^Illegal user (\S+) from ([^ ]+)/ ) { #redhat thing
@@ -100,8 +105,10 @@
       if ( $Debug >= 5 ) {
          print STDERR "DEBUG: Found -Listening on port 22- line\n";
       }
-   }
-   elsif ( $ThisLine =~ m/^(log: )?Generating .* \w+ key\./ ) { # ssh/openssh
+   } elsif ( ($Port,$Address,$Reason) = ($ThisLine =~ /^error: Bind to port ([^ ]+) on ([^ ]+) failed: (.+).$/ )) {
+      $Temp = "$Address port $Port ($Reason)";
+      $BindFailed{$Temp}++;
+   } elsif ( $ThisLine =~ m/^(log: )?Generating .* \w+ key\./ ) { # ssh/openssh
       # Don't care about this...
       if ( $Debug >= 5 ) {
          print STDERR "DEBUG: Found -Generating RSA key- line\n";
@@ -129,6 +136,16 @@
       $sftpRequests++;
    } elsif ( $ThisLine =~ m/refused connect from (.*)$/ ) {
       $RefusedConnections{$1}++;
+   } elsif ( ($Reason) = ($ThisLine =~ /^Authentication refused: (.*)$/ ) ) {
+      $RefusedAuthentication{$Reason}++;
+   } elsif ( ($Host,$Reason) = ($ThisLine =~ /^Received disconnect from ([^ ]*): (.*)$/)) {
+      $DisconnectReceived{$Reason}{$Host}++;
+   } elsif ( ($Host) = ($ThisLine =~ /^ROOT LOGIN REFUSED FROM ([^ ]*)$/)) {
+      $RootLogin{$Host}++;
+   } elsif ( ($Error) = ($ThisLine =~ /^Cannot release PAM authentication\[\d\]: (.*)$/)) {
+      $PamReleaseFail{$Error}++;
+   } elsif ( $ThisLine =~ m/^error: PAM: (.*)$/) {
+      $PamReleaseFail{$Error}++;
    } elsif ( ($IP) = ($ThisLine =~ /^scanned from ([^ ]*)/) ) {
       push @Scanned, LookupIP($IP);
    } else {
@@ -149,16 +166,31 @@
    print "\nSSHD Started: " . $Starts . " Time(s)\n";
 }
 
+if (keys %RootLogin) {
+   print "\n\nWARNING!!!\n";
+   print "Refused ROOT login attempt from:\n";
+   foreach $Host (sort {$a cmp $b} keys %RootLogin) {
+      print "   $Host : $RootLogin{$Host} Time(s)\n";
+   }
+}
+
+if (keys %BindFailed) {
+   print "\nFailed to bind:\n";
+   foreach $ThisOne (sort {$a cmp $b} keys %BindFailed) {
+      print "   $ThisOne : $BindFailed{$ThisOne} Time(s)\n";
+   }
+}
+
 if ($Detail >= 10) {
    if (keys %NoRevMap) {
       print "\nCouldn't resolve these IPs:\n";
-      foreach $ThisOne (keys %NoRevMap) {
+      foreach $ThisOne (sort {$a cmp $b} keys %NoRevMap) {
          print "   " . $ThisOne . ": " . $NoRevMap{$ThisOne} . " Time(s)\n";
       }
    }
    if (keys %NoIdent) {
       print "\nDidn't receive an ident from these IPs:\n";
-      foreach $ThisOne (keys %NoIdent) {
+      foreach $ThisOne (sort {$a cmp $b} keys %NoIdent) {
          print "   " . $ThisOne . ": " . $NoIdent{$ThisOne} . " Time(s)\n";
       }
    }
@@ -171,6 +203,13 @@
    }
 }
 
+if (keys %TooManyFailures) {
+   print "\nDisconnecting after too many authentication failures for user:\n";
+   foreach $User (sort {$a cmp $b} keys %TooManyFailures) {
+      print "   $User : $TooManyFailures{$User} Time(s)\n";
+   }
+}
+
 if (keys %BadLogins) {
    print "\nFailed logins from these:\n";
    for (sort keys %BadLogins) {
@@ -187,7 +226,7 @@
 
 if (keys %Users) {
    print "\nUsers logging in through sshd:\n";
-   foreach $user (sort keys %Users) {
+   foreach $user (sort {$a cmp $b} keys %Users) {
       print "   $user:\n";
       my $totalSort = TotalCountOrder(%{$Users{$user}}, \&SortIP);
       foreach my $ip (sort $totalSort keys %{$Users{$user}}) {
@@ -209,6 +248,23 @@
    }
 }
 
+if (keys %RefusedAuthentication) {
+   print "\n\nAuthentication refused:\n";
+   foreach $Reason (sort {$a cmp $b} keys %RefusedAuthentication) {
+      print "   $Reason : $RefusedAuthentication{$Reason} Time(s)\n";
+   }
+}
+
+if (keys %DisconnectReceived) {
+   print "\n\nReceived disconnect:\n";
+   foreach $Reason (sort {$a cmp $b} keys %DisconnectReceived) {
+      print "   $Reason\n";
+      foreach $Host (sort {$a cmp $b} keys %{$DisconnectReceived{$Reason}}) {
+         print "      $Host : $DisconnectReceived{$Reason}{$Host} Time(s)\n";
+      }
+   }
+}
+
 if ($#Scanned >= 0) {
    print "\nScanned from these:\n";
    foreach $ThisOne (sort SortIP @Scanned) {
@@ -218,8 +274,15 @@
 
 if (keys %RefusedConnections) {
    print "\nRefused incoming connections:\n";
-   foreach my $badguy ( keys %RefusedConnections ) {
+   foreach my $badguy (sort {$a cmp $b} keys %RefusedConnections ) {
       print "      $badguy: " . $RefusedConnections{$badguy} . " Time(s)\n";
+   }
+}
+
+if (keys %PamReleaseFail) {
+   print "\nCannot release PAM authentication:\n";
+   foreach $Error (sort {$a cmp $b} keys %PamReleaseFail) {
+      print "   $Error : $PamReleaseFail{$Error} Time(s)\n";
    }
 }
 
