Index: scripts/services/named
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/named,v
retrieving revision 1.33
diff -u -r1.33 named
--- scripts/services/named	3 Feb 2004 04:18:55 -0000	1.33
+++ scripts/services/named	3 Feb 2004 08:39:38 -0000
@@ -49,11 +49,9 @@
       ($ThisLine =~ /^Forwarding source address is/) or
       ($ThisLine =~ /^bad referral/) or
       ($ThisLine =~ /prerequisite not satisfied/) or
-      ($ThisLine =~ /^rcvd NOTIFY/) or
-      ($ThisLine =~ /^Sent NOTIFY/) or
+      ($ThisLine =~ /^(rcvd|Sent) NOTIFY/) or
       ($ThisLine =~ /^ns_resp: TCP truncated/) or
       ($ThisLine =~ /No possible A RRs/) or
-      ($ThisLine =~ /NS points to CNAME/) or
       ($ThisLine =~ /points to a CNAME/) or
       ($ThisLine =~ /^dangling CNAME pointer/) or
       ($ThisLine =~ /^listening on/) or
@@ -71,14 +69,13 @@
       ($ThisLine =~ /^running/) or
       ($ThisLine =~ /^exiting/) or
       ($ThisLine =~ /^no longer listening/) or
-      ($ThisLine =~ /^stopping command channel on [0-9.#]/) or
       ($ThisLine =~ /^the default for the .* option is now/) or
+      ($ThisLine =~ /^stopping command channel on [0-9.#]/) or
       ($ThisLine =~ /^Malformed response from/) or
       ($ThisLine =~ /client .+#\d+: query:/) or
       # Do we really want to ignore these?
       #($ThisLine =~ /unknown logging category/) or
       ($ThisLine =~ /^could not open entropy source/) or
-      ($ThisLine =~ /\/etc\/rndc.key: file not found/) or
       ($ThisLine =~ /\/etc\/rndc.key: file not found/) or
       ($ThisLine =~ /sending notifies/) or
       # file syntax error get reported twice and are already caught below
