--- ./scripts/services/init.origi	Sat Oct 12 04:08:18 2002
+++ ./scripts/services/init	Tue Mar 25 09:32:51 2003
@@ -24,6 +24,11 @@
       chomp ($ThisLine);
       $RunLevel{$ThisLine}++;
    }
+   elsif ( $ThisLine =~ s/^Trying to re-exec init// ) {
+	   # Look for telinit executions
+	   chomp ($ThisLine);
+	   $ReExecInit++;
+   }
    else {
       # report any unmatched entries
       push @OtherList,$ThisLine;
@@ -35,6 +40,9 @@
       print "   Entered or switched to runlevel " . $Level . ": " . $RunLevel{$Level} . " Time(s)\n";
    }
 }
+if ($ReExecInit) {
+	print "\n\nRe-execs of init: $ReExecInit times\n";
+}
 
 if ($#OtherList >= 0) {
    print "\n**Unmatched Entries**\n";
