--- ./scripts/logwatch.pl~	2005-04-10 16:33:27.000000000 +0300
+++ ./scripts/logwatch.pl	2005-04-26 11:41:15.000000000 +0300
@@ -775,7 +775,7 @@
    $index_par++;
    outstart($index_par,"LOGWATCH Summary");
    if ( $outtype_html or $outtype_htmlembed ) {
-       outheader( $index_par, "\n <h2><font color=\"blue\"> LogWatch $Version ($VDate)</font></h3>\n");
+       outheader( $index_par, "\n <h2><font color=\"blue\"> LogWatch $Version ($VDate)</font></h2>\n");
    }       else {
        outline( $index_par, "\n ################### LogWatch $Version ($VDate) #################### \n");
    }
