Index: scripts/services/postfix
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/postfix,v
retrieving revision 1.3
diff -u -r1.3 postfix
--- scripts/services/postfix	18 Nov 2003 14:04:05 -0000	1.3
+++ scripts/services/postfix	22 Nov 2003 23:24:21 -0000
@@ -51,52 +45,60 @@
 $MsgsSent = 0;
 $BytesTransferred = 0;
 $FourHourReturns = 0;
+$ReturnedToSender = 0;
 
 while (defined($ThisLine = <STDIN>)) {
-    if ( ( $ThisLine =~ m/^connect/ ) or 
-	 ( $ThisLine =~ m/^disconnect/ ) or 
-	 ( $ThisLine =~ m/^reload configuration/ ) or 
-	 ( $ThisLine =~ m/^[a-zA-Z0-9]+: client/ ) or 
-	 ( $ThisLine =~ m/^[a-zA-Z0-9]+: message-id/ ) or
-	 ( $ThisLine =~ m/^[a-zA-Z0-9]+: skipped, still being delivered/ ) or
-	 ( $ThisLine =~ m/^warning: [\.0-9]+: address not listed for hostname/ ) or
-	 ( $ThisLine =~ m/^[a-zA-Z0-9]+: to\=\<.*>, relay\=.*, delay\=[0-9]+, status\=[sent|deferred]/ ) or
-	 ( $ThisLine =~ m/^warning: [\.0-9]+: hostname .* verification failed: Host not found/ ) or
-	 ( $ThisLine =~ m/^warning: no MX host for .* has a valid A record$/ ) or
-	 ( $ThisLine =~ m/^warning: numeric domain name in resource data of MX record for .*$/ )) {
-	# We don't care about these
-    }
-    elsif ( ($Bytes) = ($ThisLine =~ /^[a-zA-Z0-9]+: from=.*size=([0-9]+).*$/) ) {
-	$MsgsSent++;
-	$BytesTransferred += $Bytes;
+	if ( ( $ThisLine =~ m/^connect/ ) or 
+		( $ThisLine =~ m/^disconnect/ ) or 
+		( $ThisLine =~ m/^reload configuration/ ) or 
+		( $ThisLine =~ m/^[a-zA-Z0-9]+: client/ ) or 
+		( $ThisLine =~ m/^[a-zA-Z0-9]+: message-id/ ) or
+		( $ThisLine =~ m/^[a-zA-Z0-9]+: skipped, still being delivered/ ) or
+		( $ThisLine =~ m/^warning: [\.0-9]+: address not listed for hostname/ ) or
+		( $ThisLine =~ m/^[a-zA-Z0-9]+: to\=\<.*>, relay\=.*, delay\=[0-9]+, status\=[sent|deferred]/ ) or
+		( $ThisLine =~ m/^warning: [\.0-9]+: hostname .* verification failed: Host not found/ ) or
+		( $ThisLine =~ m/^warning: no MX host for .* has a valid A record$/ ) or
+		( $ThisLine =~ m/^warning: numeric domain name in resource data of MX record for .*$/ )) {
+		# We don't care about these
+	}
+	elsif ( ($Bytes) = ($ThisLine =~ /^[a-zA-Z0-9]+: from=.*size=([0-9]+).*$/) ) {
+		$MsgsSent++;
+		$BytesTransferred += $Bytes;
     }
     elsif (($User) = ($ThisLine =~ /^[a-zA-Z0-9]+: to\=\<([^ ]*)>, relay\=local, delay\=[0-9]+, status\=bounced \(unknown user/)) {
-	# unknown user
-        $UnknownUsers{$User}++;
+		# unknown user
+		$UnknownUsers{$User}++;
     }
     elsif (($Dest, $Relay, $Msg) = ($ThisLine =~ /^[a-zA-Z0-9]+: to\=\<([^ ]*)>, relay=([^ ]*).*, delay\=[0-9]+, status\=bounced \(([^)]*)/ )) {
-	# unknown user
-	#$Msg = " hello "
-	#print "bounce message from " . $Dest . " msg : " . $Relay . "\n";
-	if ($Relay =~ m/^[none|local|avcheck]/) {
-	        $Temp = "To " . $Dest . " Msg=\"" . $Msg . "\"";
-        	$LocalBounce{$Temp}++;				
-	} else {
-	        $Temp = "To " . $Dest . " Msg=\"" . $Msg . "\"";
-        	$ForeignBounce{$Temp}++;
-	}
-    }
-    elsif ( ($Relay,$Dest) = ($ThisLine =~ m/^reject: RCPT from ([^ ]*): 554 <([^ ]*)>.* Relay access denied.* to=([^ ]*)/) ) {
-    	# print "reject: " . $ThisLine . "\n";
-	# print "Relay :" . $Relay . " to " . $Dest . "\n";  
-        $Temp = "From " . $Relay . " to " . $Dest;
-        $RelayDenied{$Temp}++;
+		# unknown user
+		# $Msg = " hello "
+		# print "bounce message from " . $Dest . " msg : " . $Relay . "\n";
+		if ($Relay =~ m/^[none|local|avcheck]/) {
+			$Temp = "To " . $Dest . " Msg=\"" . $Msg . "\"";
+			$LocalBounce{$Temp}++;				
+		} else {
+			$Temp = "To " . $Dest . " Msg=\"" . $Msg . "\"";
+			$ForeignBounce{$Temp}++;
+		}
+	}
+	elsif ( ($Relay,$Dest) = ($ThisLine =~ m/^reject: RCPT from ([^ ]*): 554 <([^ ]*)>.* Relay access denied.* to=([^ ]*)/) ) {
+		# print "reject: " . $ThisLine . "\n";
+		# print "Relay :" . $Relay . " to " . $Dest . "\n";
+		$Temp = "From " . $Relay . " to " . $Dest;
+		$RelayDenied{$Temp}++;
     }
     elsif ( ($User,$From) = ($ThisLine =~ /^[a-zA-Z0-9]+: uid=([^ ]*) from=\<([^ ]*)>/)) {
 		#Messages sent by user
 		$Temp = $From . " (uid=" . $User . "): ";
 		$SentBy{$Temp}++;
 	}
+	elsif ( ($From) = ($ThisLine =~ /^[a-zA-Z0-9]+: from=<([^ ]*)>, status=expired, returned to sender$/)) {
+		$ReturnedToSender++;
+	}
+	elsif ( ($Command,$Host) = ($ThisLine =~ /lost connection after ([^ ]*) from ([^ ]*)$/)) {
+		# Make some better summary with hosts
+		$ConnectionLost{$Command}++;
+	}
 #    elsif ( ($Warning)  = ($ThisLine =~ /^[a-zA-Z0-9]+: Authentication-Warning: [^ ]+: ([^ ]+ set sender to [^ ]+ using -f)/) ) {
 #        $AuthWarns{$Warning}++;
 #    }
@@ -118,15 +120,18 @@
 if ((@OtherList) or
     ($MsgsSent > 0) or
     ($FourHourReturns > 0) or
+	($ReturnedToSender >0) or
     (keys %UnknownUsers) or
     (keys %RelayDenied) or
     (keys %ForeignBounce) or
     (keys %LocalBounce) or
     (keys %AuthWarns) or
     (keys %ForwardErrors) or
-	(keys %SentBy)) {
-    print "\n\n --------------------- postfix Begin ------------------------ ";
-
+	(keys %SentBy) or
+	(keys %ConnectionLost)
+) {
+	print "\n\n --------------------- postfix Begin ------------------------ ";
+	
     if($MsgsSent > 0) {
         print "\n\n" . $BytesTransferred . " bytes transferred";
         print "\n" . $MsgsSent . " messages sent";
@@ -136,6 +141,10 @@
         print "\n\n" . $FourHourReturns . " messages returned after 4 hours";
     }
 
+	if ($ReturnedToSender >0) {
+		print "\n\n" . $ReturnedToSender . " messages expired and returned to sender";
+	}
+
 	if (keys %SentBy) {
 		print "\n\nMessages sent by:\n";
 		foreach $ThisOne (keys %SentBy) {
@@ -151,45 +160,52 @@
 	}
 
     if (keys %RelayDenied) {
-	print "\n\nRelaying denied:\n";
-	foreach $ThisOne (keys %RelayDenied) {
-	    print "    " . $ThisOne . ": " . $RelayDenied{$ThisOne} . " Times(s)\n";
-	}
+		print "\n\nRelaying denied:\n";
+		foreach $ThisOne (keys %RelayDenied) {
+			print "    " . $ThisOne . ": " . $RelayDenied{$ThisOne} . " Times(s)\n";
+		}
     }
 
     if (keys %LocalBounce) {
-	print "\n\nLocal Bounce:\n";
-	foreach $ThisOne (keys %LocalBounce) {
-	    print "    " . $ThisOne . ": " . $LocalBounce{$ThisOne} . " Times(s)\n";
-	}
+		print "\n\nLocal Bounce:\n";
+		foreach $ThisOne (keys %LocalBounce) {
+			print "    " . $ThisOne . ": " . $LocalBounce{$ThisOne} . " Times(s)\n";
+		}
     }
 
     if (keys %ForeignBounce) {
-	print "\n\nForeign Bounce:\n";
-	foreach $ThisOne (keys %ForeignBounce) {
-	    print "    " . $ThisOne . ": " . $ForeignBounce{$ThisOne} . " Times(s)\n";
-	}
+		print "\n\nForeign Bounce:\n";
+		foreach $ThisOne (keys %ForeignBounce) {
+			print "    " . $ThisOne . ": " . $ForeignBounce{$ThisOne} . " Times(s)\n";
+		}
     }
 
     if (keys %AuthWarns) {
-	print "\n\nAuthentication warnings:\n";
-	foreach $ThisOne (keys %AuthWarns) {
-	    print "    " . $ThisOne . ": " . $AuthWarns{$ThisOne} . " Times(s)\n";
-	}
+		print "\n\nAuthentication warnings:\n";
+		foreach $ThisOne (keys %AuthWarns) {
+			print "    " . $ThisOne . ": " . $AuthWarns{$ThisOne} . " Times(s)\n";
+		}
     }
 
     if (keys %ForwardErrors) {
-	print "\n\nForwarding errors:\n";
-	foreach $ThisOne (keys %ForwardErrors) {
-	    print "    " . $ThisOne . ": " . $ForwardErrors{$ThisOne} . " Times(s)\n";
+		print "\n\nForwarding errors:\n";
+		foreach $ThisOne (keys %ForwardErrors) {
+			print "    " . $ThisOne . ": " . $ForwardErrors{$ThisOne} . " Times(s)\n";
+		}
 	}
-    }
 
+	if (keys %ConnectionLost) {
+		print "\n\nConnections lost:\n";
+		foreach $ThisOne (keys %ConnectionLost) {
+			print "    Connection lost after command " . $ThisOne . ": " . $ConnectionLost{$ThisOne} . " Times(s)\n";
+		}
+	}
+	
     if ($#OtherList >= 0) {
-	print "\n\n**Unmatched Entries**\n\n";
-	print @OtherList;
-    }
-
+		print "\n\n**Unmatched Entries**\n\n";
+		print @OtherList;
+	}
+	
     print "\n ---------------------- postfix End ------------------------- \n";
 
 }
