Index: conf/services/pop3.conf
===================================================================
RCS file: /var/cvs/logwatch/conf/services/pop3.conf,v
retrieving revision 1.1
diff -u -r1.1 pop3.conf
--- conf/services/pop3.conf	21 Jun 2004 14:59:04 -0000	1.1
+++ conf/services/pop3.conf	2 Oct 2004 14:45:09 -0000
@@ -20,7 +20,7 @@
 # Only give lines pertaining to courier...
 # I'm not sure if this is complete, especially for the new webmail daemon in 0.44.1
 #but you will get at least all currently supported logs 
-*OnlyService = (pop3d-ssl|pop3d|spop3d)
+*OnlyService = (pop3d-ssl|pop3d|spop3d|tpop3d)
 
 *RemoveHeaders = 
 
