Index: conf/logfiles/clam-update.conf
===================================================================
RCS file: /var/cvs/logwatch/conf/logfiles/clam-update.conf,v
retrieving revision 1.2
diff -u -r1.2 clam-update.conf
--- conf/logfiles/clam-update.conf	16 Dec 2003 14:21:11 -0000	1.2
+++ conf/logfiles/clam-update.conf	28 Apr 2004 11:49:43 -0000
@@ -43,6 +43,7 @@
 
 # What actual file?  Defaults to LogPath if not absolute path....
 LogFile = clam-update
+LogFile = freshclam.log
 
 # If the archives are searched, here is one or more line
 # (optionally containing wildcards) that tell where they are...
@@ -50,4 +51,10 @@
 #       you use wildcards...
 Archive = clam-update.*
 Archive = clam-update.*.gz
+Archive = freshclam.log.*
+Archive = freshclam.log.*.gz
+Archive = archiv/clam-update.*
+Archive = archiv/clam-update.*.gz
+Archive = archiv/freshclam.log.*
+Archive = archiv/freshclam.log.*.gz
 
