Index: scripts/services/clam-update
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/clam-update,v
retrieving revision 1.5
diff -u -r1.5 clam-update
--- scripts/services/clam-update	3 Feb 2004 04:39:56 -0000	1.5
+++ scripts/services/clam-update	20 May 2004 06:18:39 -0000
@@ -1,5 +1,7 @@
 #!/usr/bin/perl -w
-
+##########################################################################
+## $Id$
+###########################################################################
 #########################################################################
 # clam-update script for Logwatch
 # Analyzes the Clam Anti-Virus update log
@@ -69,8 +71,9 @@
 my $DB2Uptodate   = undef;
 my $DB2Updated    = undef;
 
-my $bDBUptodate   = 0;
-my $bDB2Uptodate  = 0;
+my $bDBUptodate      = 0;
+my $bDB2Uptodate     = 0;
+my $NotificationOK   = 0;
 
 my %Errors;
 my %Unmatched;
@@ -126,6 +129,8 @@
       } elsif ($ThisLine =~ /^viruses\.db2 is up to date/) {
          $bDB2Uptodate = 1;
          $DB2Uptodate  = $ThisLine;
+      } elsif ($ThisLine =~ /^Clamd successfully notified about the update./) {
+         $NotificationOK++;
       } elsif (($Text) = ($ThisLine =~ /^ERROR: (.*)/)) {
          $Errors{$Text}++;
       } else {
@@ -148,6 +153,8 @@
    }
 }
 
+
+#####################################################################
 # This should not be necessary since a header will be inserted by the
 # main logwatch program if output is genereated - Kirk
 #if (($Detail >= 5) or ($MainUpdated or $DailyUpdated or $DBUpdated or $DB2Updated) or (!$MainUptodate and !$DailyUptodate and !$DBUptodate and !$DB2Uptodate)) {
@@ -208,6 +215,13 @@
    }
 }
 
+if ($NotificationOK > 0) {
+   print "Clamd successfully notified about the update $NotificationOK Time(s).\n";
+} elsif (($MainUpdated or $DailyUpdated) and ($NotificationOK > 0)) {
+   print "WARNING\n";
+   print "Databases are updated, but Clamd is not notified.\n";
+}
+
 if (($Updated) and ($Detail >= 10)) {
    if ($Updated =~ /^(\w* \w*) \(\d* \w*\)/) {
       (my $Text, my $From) = ($Updated =~ /^(\w* \w*) \(\d* \w*\) (.*)\./);
@@ -238,3 +252,5 @@
 }
 
 exit(0);
+
+# vi: shiftwidth=3 tabstop=3 et
