Index: scripts/services/amavis
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/amavis,v
retrieving revision 1.13
diff -u -r1.13 amavis
--- scripts/services/amavis	29 Oct 2004 21:19:00 -0000	1.13
+++ scripts/services/amavis	31 Oct 2004 16:10:00 -0000
@@ -65,7 +65,7 @@
        # We don't care about these
    } elsif ($ThisLine =~ /^Passed,? /) {
       $CleanMsgs++;
-   } elsif (($FileName, $From) = ( $ThisLine =~ /^(?:Blocked )?BANNED (?:name\/type )?\(([^\)]+)\)\, (?:\[([^\]]*)\] )?\<([^\>]*)\>/ )) {
+   } elsif (($FileName, $From) = ( $ThisLine =~ /^(?:Blocked )?BANNED (?:name\/type )?\(([^\)]+)\)\, (?:\[([^\]]*)\] )*\<([^\>]*)\>/ )) {
 
       $BannedNames++;
 
@@ -73,7 +73,7 @@
           $Banned{$FileName}{$From}++;
       }; # if 
 
-   } elsif (($Virus, $FromIP, $From) = ( $ThisLine =~ /^(?:Virus found - quarantined|INFECTED|Blocked INFECTED) \(([^\)]+)\)\, (?:\[([^\]]*)\] )?[\(\<]([^\>\)]*)[\)\>]/ )) {
+   } elsif (($Virus, $FromIP, $From) = ( $ThisLine =~ /^(?:Virus found - quarantined|INFECTED|Blocked INFECTED) \(([^\)]+)\)\, (?:\[([^\]]*)\] )*[\(\<]([^\>\)]*)[\)\>]/ )) {
 
       $InfectedMsgs++;
 
@@ -92,7 +92,7 @@
         $Viruses{$Virus}{$From}++;
       }; # if
 
-   } elsif (($FromIP, $Fromspam, $Towards) = ( $ThisLine =~ /^(?:SPAM|Blocked SPAM), (?:\[([^\]]*)\] )?[\(\<]([^\>\)]+)[\)\>] -\> [\(\<]([^\>\)]+)[\)\>]/ )) {
+   } elsif (($FromIP, $Fromspam, $Towards) = ( $ThisLine =~ /^(?:SPAM|Blocked SPAM), (?:\[([^\]]*)\] )*[\(\<]([^\>\)]+)[\)\>] -\> [\(\<]([^\>\)]+)[\)\>]/ )) {
       $SpamMsgs++;
 
       if (defined($Fromspam)){
@@ -121,7 +121,7 @@
         $Spams{$Towards}{"<>"}++;
       }; # if
 
-   } elsif (($FromIP, $From, $Why) = ( $ThisLine =~ /^(?:Blocked )?BAD[ -]HEADER(?: from|\,) (?:\[([^\]]*)\] )?[\(\<]([^\>\)]+)[\)\>](?: -\> [\(\<](?:[^\>\)]+)[\)\>])[^\:]*(?:\: (.*) in message header)?/ )) {
+   } elsif (($FromIP, $From, $Why) = ( $ThisLine =~ /^(?:Blocked )?BAD[ -]HEADER(?: from|\,) (?:\[([^\]]*)\] )*[\(\<]([^\>\)]+)[\)\>](?: -\> [\(\<](?:[^\>\)]+)[\)\>])[^\:]*(?:\: (.*) in message header)?/ )) {
       $BadHeaders++;
 
       if (defined($From)){
