--- ./scripts/services/sendmail.warning	Tue Feb 18 16:42:01 2003
+++ ./scripts/services/sendmail	Fri Mar 28 11:48:33 2003
@@ -84,7 +84,7 @@
       $DomainErrors{$User . " (does not exist)"}++;
    } elsif ( ($User) = ($ThisLine =~ /^ruleset=check_mail, arg1=<([^ ]*)>, relay=[^,]*, reject=553\s*[\d.]*\s*<[^ ]*>\.\.\. Domain name required for sender address .*/) ) {
       $DomainErrors{$User . " (missing)"}++;
-   } elsif ( ($Warning)  = ($ThisLine =~ /Authentication-Warning: [^ ]+: ([^ ]+ set sender to [^ ]+ using -f|.+ didn\'t use HELO protocol)/) ) {
+   } elsif ( ($Warning)  = ($ThisLine =~ /Authentication-Warning: [^ ]+: ([^ ]+ set sender to  [^ ]+ using -f|.+ didn\'t use HELO protocol|[^ ]+ owned process doing -bs)/) ) {
       $AuthWarns{$Warning}++;
    } elsif ( ($Forward,$Error) = ($ThisLine =~ /^forward ([^ ]*): transient error: (.*)$/) ) {
       $Temp = $Forward . ": " . $Error;
