Index: scripts/services/postfix
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/postfix,v
retrieving revision 1.14
diff -u -r1.14 postfix
--- scripts/services/postfix	29 Oct 2004 21:19:01 -0000	1.14
+++ scripts/services/postfix	31 Oct 2004 15:07:36 -0000
@@ -123,12 +123,14 @@
       ( $ThisLine =~ m/^[a-zA-Z0-9]+: skipped, still being delivered/ ) or
       ( $ThisLine =~ m/^[a-zA-Z0-9]+: to\=\<.*>, relay\=.*, delay\=[0-9]+, status\=(sent|deferred)/ ) or
       ( $ThisLine =~ m/^[a-zA-Z0-9]+: host [^ ]*\[[^ ]*\] said: 4[0-9][0-9]/ ) or
+      ( $ThisLine =~ m/^[a-zA-Z0-9]+: host [^ ]*\[[^ ]*\] refused to talk to me: 4[0-9][0-9]/ ) or
       ( $ThisLine =~ m/^Deleted: \d message$/ ) or
       ( $ThisLine =~ m/^Peer certficate could not be verified$/ ) or #postfix typo
       ( $ThisLine =~ m/^Peer certificate could not be verified$/ ) or
       ( $ThisLine =~ m/^Peer verification:/ ) or
       ( $ThisLine =~ m/^SSL_accept error from/ ) or
       ( $ThisLine =~ m/^Verified: / ) or
+      ( $ThisLine =~ m/^cert has expired/ ) or
       ( $ThisLine =~ m/^connect/ ) or 
       ( $ThisLine =~ m/^daemon started$/ ) or
       ( $ThisLine =~ m/^daemon started -- version / ) or
