Index: conf/services/secure.conf
===================================================================
RCS file: /var/cvs/logwatch/conf/services/secure.conf,v
retrieving revision 1.10
diff -u -r1.10 secure.conf
--- conf/services/secure.conf	18 Oct 2002 18:58:01 -0000	1.10
+++ conf/services/secure.conf	2 Oct 2004 12:52:55 -0000
@@ -24,7 +24,7 @@
 # Use this to ignore certain services in the secure log.
 # You can ignore as many services as you would like.
 # (we ignore sshd because its entries are processed by the sshd script)
-$ignore_services = sshd Pluto stunnel proftpd
+$ignore_services = sshd Pluto stunnel proftpd saslauthd
 
 # For these services, summarize only (i.e. don't least each IP, just
 # list the number of connections total)
