Index: scripts/services/http
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/http,v
retrieving revision 1.17
diff -u -r1.17 http
--- scripts/services/http	24 Feb 2005 22:51:45 -0000	1.17
+++ scripts/services/http	12 Mar 2005 22:32:12 -0000
@@ -110,6 +110,10 @@
 my $fonts_bytes =    0;
 my $config_count =   0;
 my $config_bytes =   0;
+my $xpcomext_count = 0;
+my $xpcomext_bytes = 0;
+my $mozext_count =   0;
+my $mozext_bytes =   0;
 my $proxy_count =    0;
 my $proxy_bytes =    0;
 my %proxy_host =     ();
@@ -125,22 +129,24 @@
    $content_types =  $content_types.'|^\/announce|^\/scrape'; # BitTorrent tracker mod_bt
    $content_types =  $content_types.'|\.torrent';
    $content_types =  $content_types.'|\.css|\.js|\.cgi';
-   $content_types =  $content_types.'|\.fla|\.swf';
+   $content_types =  $content_types.'|\.fla|\.swf|\.rdf';
    $content_types =  $content_types.'|\.class|\.jsp|\.jar|\.java';
    $content_types =  $content_types.'|COPYRIGHT|readme|README|FAQ|INSTALL|\.txt)';
-my $docs_types =     '(\.asc|\.djvu|\.doc|\.dot|\.dvi|\.gnumeric|\.mcd|\.mso|\.pdf|\.pps|\.ppt|\.ps|\.rtf|\.sxi|\.tex|\.text|\.tm|\.xls|\.xml)';
+my $docs_types =     '(\.asc|\.djvu|\.doc|\.dot|\.dtd|\.dvi|\.gnumeric|\.mcd|\.mso|\.pdf|\.pps|\.ppt|\.ps|\.rtf|\.sxi|\.tex|\.text|\.tm|\.xls|\.xml)';
 my $archive_types =  '(\.ace|\.bz2|\.cab|\.deb|\.dsc|\.ed2k|\.gz|\.hqx|\.md5|\.rar|\.rpm|\.sig|\.sign|\.tar|\.tbz2|\.tgz|\.vl2|\.Z|\.zip)';
-my $sound_types =    '(\.au|\.mid|\.mp3|\.ogg|\.pls|\.ram|\.raw|\.rm|\.wav|\.wma|\.wmv|\.xsm)';
+my $sound_types =    '(\.au|\.aud|\.mid|\.mp3|\.ogg|\.pls|\.ram|\.raw|\.rm|\.wav|\.wma|\.wmv|\.xsm)';
 my $movie_types =    '(\.asf|\.ass|\.avi|\.idx|\.mid|\.mpg|\.mpeg|\.mov|\.qt|\.psb|\.srt|\.ssa|\.smi|\.sub)';
 my $winexec_types =  '(\.bat|\.com|\.exe|\.dll)';
 my $wpad_files =     '(wpad\.dat|wspad\.dat|proxy\.pac)';
 my $program_src =    '(';
-   $program_src =    $program_src.'\.bas|\.c|\.cpp|\.diff|\.f|\.h|\.init|\.m|\.pas|\.patch|\.spec';
+   $program_src =    $program_src.'\.bas|\.c|\.cpp|\.diff|\.f|\.h|\.init|\.m|\.mo|\.pas|\.patch|\.po|\.pot|\.sh|\.spec';
    $program_src =    $program_src.'|Makefile|Makefile_c|Makefile_f77)';
 my $images_types =   '(\.bin|\.cue|\.img|\.iso|\.run)';
 my $logs_types =     '(\.log|_log|-log|\.logs|\.out|\.wyniki)';
 my $fonts_types =    '(\.aft|\.ttf)';
-my $config_types =   '(\.cfg|\.conf|\.config|\.ini)';
+my $config_types =   '(\.cfg|\.conf|\.config|\.ini|\.properties)';
+my $xpcomext_types = '(\.xpt)';
+my $mozext_types =   '(\.xul)';
 
 #
 #   what to look for as an attack  USE LOWER CASE!!!!!!
@@ -309,6 +315,12 @@
    } elsif ($field{base_url} =~ /$config_types$/) {
       $config_count += 1;
       $config_bytes += $field{bytes_transfered};
+   } elsif ($field{base_url} =~ /$xpcomext_types$/) {
+      $xpcomext_count += 1;
+      $xpcomext_bytes += $field{bytes_transfered};
+   } elsif ($field{base_url} =~ /$mozext_types$/) {
+      $mozext_count += 1;
+      $mozext_bytes += $field{bytes_transfered};
    } elsif ($field{http_rc} =~ /3\d\d/) {
       $redirect_count += 1;
       $redirect_bytes += $field{bytes_transfered};
@@ -382,6 +394,8 @@
    if ($logs_count > 0)       { printf " $logs_count various Logs (%.2f MB),\n" , $logs_bytes/(1024*1024); }
    if ($fonts_count > 0)      { printf " $fonts_count Fonts (%.2f MB),\n" , $fonts_bytes/(1024*1024); }
    if ($config_count > 0)     { printf " $config_count Configs (%.2f MB),\n" , $config_bytes/(1024*1024); }
+   if ($xpcomext_count > 0)   { printf " $xpcomext_count XPCOM Type Libraries (%.2f MB),\n" , $xpcomext_bytes/(1024*1024); }
+   if ($mozext_count > 0)     { printf " $mozext_count Mozilla extensions (%.2f MB),\n" , $mozext_bytes/(1024*1024); }
    if ($proxy_count > 0)      { printf " $proxy_count mod_proxy connection attempts (%.2f MB),\n" , $proxy_bytes/(1024*1024); }
    if ($other_count > 0)      { printf " $other_count Other (%.2f MB) \n" , $other_bytes/(1024*1024); }
 }
