Index: scripts/services/samba
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/samba,v
retrieving revision 1.17
diff -u -r1.17 samba
--- scripts/services/samba	15 Dec 2003 22:03:55 -0000	1.17
+++ scripts/services/samba	18 Dec 2003 07:30:44 -0000
@@ -2,6 +2,8 @@
 ##########################################################################
 # $Id$
 ##########################################################################
+# $Log$
+##########################################################################
 
 ########################################################
 # This was written and is maintained by:
@@ -107,6 +109,9 @@
 		$RootLoggedIn{$user}++;
 	} elsif ( ($file,$function) = ($ThisLine =~ /([a-zA-Z_\/():\.0-9-]+)  ([a-zA-Z0-9_-]+): Not yet implemented.$/)) {
 		$NotImplemented{$file}{$function}++;
+   } elsif ( ($User,$Ip,$Directory,$Reason) = ($ThisLine =~ /service.c:make_connection\(599\)  ([^ ]+) \(([^ ]+)\) Can't change directory to ([^ ]+) \(([^ ]+)\)/)) {
+      $Temp = "Netbios name $User on $Ip";
+      $CantChangeDir{$Directory}{$Reason}{$Temp}++;
 	} else {
 		# Report any unmatched entries...
 		$OtherList{$ThisLine}++;
@@ -291,6 +296,19 @@
     foreach $User (sort {$a cmp $b} keys %AccountDisabled) {
         print "   $User : $AccountDisabled{$User} Time(s)\n";
     }
+}
+
+if (keys %CantChangeDir) {
+   print "\nCan't change directory while browsing:\n";
+   foreach $Directory (sort {$a cmp $b} keys %CantChangeDir) {
+      print "   $Directory:\n";
+      foreach $Reason (sort {$a cmp $b} keys %{$CantChangeDir{$Directory}}) {
+         print "      $Reason:\n";
+         foreach $Entry (sort {$a cmp $b} keys %{$CantChangeDir{$Directory}{$Reason}}) {
+            print "         $Entry : $CantChangeDir{$Directory}{$Reason}{$Entry} Time(s)\n";
+         }
+      }
+   }
 }
 
 if ($SocketReadError > 0) {
