--- ./conf/logfiles/clam-update.conf.origin	Tue Dec  2 08:52:26 2003
+++ ./conf/logfiles/clam-update.conf	Tue Dec  2 08:54:05 2003
@@ -38,12 +38,12 @@
 #########################################################################
 
 # What actual file?  Defaults to LogPath if not absolute path....
-LogFile = clam-update
+LogFile = clamav.log
 
 # If the archives are searched, here is one or more line
 # (optionally containing wildcards) that tell where they are...
 # Note: if these are gzipped, you need to end with a .gz even if
 #       you use wildcards...
-Archive = clam-update.*
-Archive = clam-update.*.gz
+Archive = clamav.log.*
+Archive = clamav.log.*.gz
 
--- ./conf/logfiles/messages.conf.origin	Tue Dec  2 08:55:10 2003
+++ ./conf/logfiles/messages.conf	Tue Dec  2 08:56:31 2003
@@ -17,8 +17,8 @@
 # (optionally containing wildcards) that tell where they are...
 # Note: if these are gzipped, you need to end with a .gz even if
 #       you use wildcards...
-Archive = messages.*
-Archive = messages.*.gz
+Archive = archiv/messages.*
+Archive = archiv/messages.*.gz
 
 # Expand the repeats (actually just removes them now)
 *ExpandRepeats
