--- ./scripts/services/postfix.org	Tue Nov 18 09:55:14 2003
+++ ./scripts/services/postfix	Tue Nov 18 09:54:49 2003
@@ -58,7 +56,9 @@
 	 ( $ThisLine =~ m/^[a-zA-Z0-9]+: skipped, still being delivered/ ) or
 	 ( $ThisLine =~ m/^warning: [\.0-9]+: address not listed for hostname/ ) or
 	 ( $ThisLine =~ m/^[a-zA-Z0-9]+: to\=\<.*>, relay\=.*, delay\=[0-9]+, status\=[sent|deferred]/ ) or
-	 ( $ThisLine =~ m/^warning: [\.0-9]+: hostname .* verification failed: Host not found/ ) ) {
+	 ( $ThisLine =~ m/^warning: [\.0-9]+: hostname .* verification failed: Host not found/ ) or
+	 ( $ThisLine =~ m/^warning: no MX host for .* has a valid A record$/ ) or
+	 ( $ThisLine =~ m/^warning: numeric domain name in resource data of MX record for .*$/ )) {
 	# We don't care about these
     }
     elsif ( ($Bytes) = ($ThisLine =~ /^[a-zA-Z0-9]+: from=.*size=([0-9]+).*$/) ) {
