diff -dur logwatch-5.2.2.orig/scripts/services/amavis logwatch-5.2.2/scripts/services/amavis
--- logwatch-5.2.2.orig/scripts/services/postfix	2004-06-21 16:59:05.000000000 +0200
+++ logwatch-5.2.2/scripts/services/postfix	2004-09-20 15:02:58.978112984 +0200
@@ -105,6 +105,7 @@
 $ErrorRBL = 0;
 $NoFreeSpace = 0;
 $RejectClients = 0;
+$RejectUnknownClients = 0;
 
 while (defined($ThisLine = <STDIN>)) {
    if ( 
@@ -208,6 +209,10 @@
    } elsif ( ($Host) = ($ThisLine =~ /reject: RCPT from ([^ ]*\[[^ ]*\]): [0-9]+ <[^ ]*\[[^ ]*\]>: Client host rejected: Access denied;/)) {
       $RejectClientHost{$Host}++;
       $RejectClients++;
+   } elsif ( ($Host,$Sender,$Recip,$Helo) = ($ThisLine =~ /reject: RCPT from [^ ]*\[([^ ]*)\]: [0-9]+ Client host rejected: cannot find your hostname, \[\d+\.\d+\.\d+\.\d+\]; from=<(.*?)> to=<(.*?)> proto=\S+ helo=<(.*)>/)) {
+      $RejectUnknownClient{$Host}{$Helo}{$Sender}{$Recip}++;
+      $RejectUnknownClientHost{"$Host	helo=<$Helo>"}++;
+      $RejectUnknownClients++;
    } elsif ( ($Host,$Recip,$Reason) = ($ThisLine =~ /reject: RCPT from ([^ ]*\[[^ ]*\]): [0-9]+ <(.*)>: Recipient address rejected: (.*);/)) {
       $Temp = "$Host : $Reason";
       $RejectRecip{$Recip}{$Temp}++;
@@ -677,12 +682,42 @@
    else {
       $n=0;
       foreach $Host (keys %RejectClientHost) {
-         $n+=$RejectClientHost{$Host};
+         $n++;
       }
       print "\n\nClient hosts rejected $RejectClients Time(s): $n Host(s)";
    }
 }
 
+if ($RejectUnknownClients > 0) {
+   if ($Detail >= 10) {
+      print "\n\nUnknown client hosts rejected $RejectUnknownClients Time(s)\n";
+      foreach $Host (sort {$a cmp $b} keys %RejectUnknownClient) {
+         print "   $Host\n";
+	 foreach $Helo (sort {$a cmp $b} keys %{$RejectUnknownClient{$Host}}) {
+	    print "	helo=<$Helo>\n";
+               foreach $Sender (sort {$a cmp $b} keys %{$RejectUnknownClient{$Host}{$Helo}}) {
+                  foreach $Recip (sort {$a cmp $b} keys %{$RejectUnknownClient{$Host}{$Helo}{$Sender}}) {
+                     print "		$Sender -> $Recip	$RejectUnknownClient{$Host}{$Helo}{$Sender}{$Recip} Time(s)\n";
+	       }
+            }
+	 }
+      }
+   }
+   elsif ($Detail >= 5) {
+      print "\n\nUnknown client hosts rejected $RejectUnknownClients Time(s)\n";
+      foreach $Host (sort {$a cmp $b} keys %RejectUnknownClientHost) {
+         print "   $Host	$RejectUnknownClientHost{$Host} Time(s)\n";
+      }
+   }
+   else {
+      $n=0;
+      foreach $Host (keys %RejectUnknownClientHost) {
+         $n++;
+      }
+      print "\n\nUnknown client hosts rejected $RejectUnknownClients Time(s): $n Host(s)";
+   }
+}
+
 if (keys %RejectSender) {
    print "\n\nMessages rejected:\n";
    foreach $Reason (sort {$a cmp $b} keys %RejectSender) {
