--- ./scripts/logwatch.pl~	2005-04-26 11:49:17.000000000 +0300
+++ ./scripts/logwatch.pl	2005-04-26 12:16:02.000000000 +0300
@@ -757,7 +757,6 @@
 	       $out_mime.=  "From: LogWatcher\n";
 	       $out_mime.=  "Subject: LogWatch for " . $Config{'hostname'} . "\n";
 	       $out_mime.= "MIME-Version: 1.0\n"; 
-          $out_mime.= "Content-Transfer-Encoding: quoted-printable\n";
           $out_mime.= "Content-Type: text/html; charset=\"iso-8859-1\"\n\n"
 
 
