Index: scripts/services/samba
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/samba,v
retrieving revision 1.19
diff -u -r1.19 samba
--- scripts/services/samba	23 Jun 2004 15:01:17 -0000	1.19
+++ scripts/services/samba	15 Jul 2004 12:07:36 -0000
@@ -25,6 +25,7 @@
 $SocketReadError = 0;
 $SocketWriteError = 0;
 $DbOpenFail = 0;
+$DbCorrupt = 0;
 $GetDomainMasterStatusFail = 0;
 
 if ( $Debug >= 5 ) {
@@ -92,6 +93,10 @@
 		$InvalidUser{$User}++;
 	} elsif (($User) = $ThisLine =~ /Couldn't find user '([^ ]+)'/) {
 		$NotFoundUser{$User}++;
+   } elsif (($User) = $ThisLine =~ /pdb_smbpasswd\.c:build_sam_account\(\d+\)  build_sam_account: smbpasswd database is corrupt!  username ([^ ]+) not in unix passwd database!/ ) {
+      $NotFoundUser{$User}++;
+      $DbOpenFail++;
+      $DbCorrupt++;
 	} elsif (($User) = $ThisLine =~ /Rejecting user '([^ ]+)'/) {
 		$RejectedUser{$User}++;
 	} elsif ( ( $ThisLine =~ /lib\/util_sock.c:read_data\(436\)/ ) ) {
@@ -104,8 +109,15 @@
    ) {
       # Something more generic should be here
       $SocketWriteError++;
-	} elsif ( ( $ThisLine =~ /unable to open passdb database.$/ ) ) {
+	} elsif (
+      ( $ThisLine =~ /unable to open passdb database.$/ ) or
+      ( $ThisLine =~ / get_sampwd_entries: Unable to open passdb.$/ )
+   ) {
 		$DbOpenFail++;
+   } elsif ( ($Name,$Group,$Subnet) = ( $ThisLine =~ /pdb_smbpasswd.c:pdb_getsampwnam\(\d+\)  unable to open passdb database\.  [ *]+Samba name server ([^ ]+) is now a local master browser for workgroup ([^ ]+) on subnet ([^ ]+)   / )) {
+      $Temp = "$Where  ($Ip)";
+      $DbOpenFail++;
+      $BeLocalMaster{$Subnet}{$Group}{$Name}++;
 	} elsif ( ($Server,$Ip,$Group) = ($ThisLine =~ /Server ([^ ]+) at IP ([^ ]+) is announcing itself as a local master browser for workgroup ([^ ]+) and we think we are master. Forcing election.$/ ) ) {
 		$Temp = $Server . "(" . $Ip . ")";
 		$ForceElection{$Group}{$Temp}++;
@@ -343,6 +355,10 @@
 	print "\nFailed to open passwd database: $DbOpenFail Time(s)\n";
 }
 
+if ($DbCorrupt > 0) {
+	print "smbpasswd database corrupt: $DbOpenFail Time(s)\n";
+}
+
 if (keys %InvalidUser) {
    print "\nInvalid Users:\n";
    foreach $Line (sort {$a cmp $b} keys %InvalidUser) {
