--- ./scripts/services/pam_unix.org	Sat Oct 12 04:08:19 2002
+++ ./scripts/services/pam_unix	Fri Mar 21 10:36:15 2003
@@ -60,6 +60,41 @@
       } else {
          $data{$service}{'Unknown Entries'}{$line}++;
       }
+   } elsif ($service eq 'imap') {
+	   if ($line =~ s/^authentication failure; logname=(.+) uid=(\d+) .*user=(.+)$/$1($2) -> $3/) {
+		   $data{$service}{'Authentication Failures'}{$line}++;
+	   } else {
+		   $data{$service}{'Unknown Entries'}{$line}++;
+	   }
+   } elsif ($service eq 'spop3d') {
+	   if ($line =~ s/^session opened for user (.+)/$1/) {
+		   $data{$service}{'Sessions Opened'}{$line}++;
+	   } elsif ($line =~ /session closed for user/) {
+		   # ignore this line
+	   } elsif ($line =~ s/^authentication failure; logname=(.+) uid=(\d+) .*user=(.+)$/$1($2) -> $3/) {
+		   $data{$service}{'Authentication Failures'}{$line}++;
+	   } else {
+		   $data{$service}{'Unknown Entries'}{$line}++;
+	   }
+   } elsif ($service eq 'cron') {
+	   if ($line =~ s/^session opened for user (.+) by \(uid=\d+\)/$1/) {
+		   ($Detail >= 5) && $data{$service}{'Sessions Opened'}{$line}++;
+	   } elsif ($line =~ /session closed for user/) {
+		   # ignore this line
+	   } else {
+		   $data{$service}{'Unknown Entries'}{$line}++;
+	   }
+   } elsif ($service eq 'pure-ftpd') {
+	   if ($line =~ s/^session opened for user (.+)/$1/) {
+		   $data{$service}{'Sessions Opened'}{$line}++;
+	   } else {
+		   $data{$service}{'Unknown Entries'}{$line}++;
+	   }
+   } elsif ($service eq 'kdm') {
+	   if ($line =~ s/^session opened for user (.+) by \(uid=\d+\)/$1/) {
+		   ($Detail >= 5) && $data{$service}{'Sessions Opened'}{$line}++;
+	   } elsif ($line =~ /session closed for user/) {
+		   # ignore this line
    } else {
       $data{$service}{'Unknown Entries'}{$line}++;
    }
--- ./scripts/services/secure.org	Tue Feb 18 16:34:53 2003
+++ ./scripts/services/secure	Fri Mar 21 10:12:36 2003
@@ -103,8 +103,12 @@
       push @NewUsers, "   $ThisLine\n";
    } elsif ( $ThisLine =~ s/^userdel\[\d+\]: remove group `(.+)'/$1/ ) {
       push @DeletedGroups, "   $ThisLine\n";
+   } elsif ( $ThisLine =~ s/^groupdel\[\d+\]: remove group `(.+)'/$1/ ) {
+      push @DeletedGroups, "   $ThisLine\n";
    } elsif ( $ThisLine =~ s/^useradd\[\d+\]: new group: name=(.+), gid=(\d+).*$/$1 ($2)/ ) {
       push @NewGroups, "   $ThisLine\n";
+   } elsif ( $ThisLine =~ s/^groupadd\[\d+\]: new group: name=(.+), gid=(\d+).*$/$1 ($2)/ ) {
+      push @NewGroups, "   $ThisLine\n";
    } elsif ( $ThisLine =~ /^userdel\[\d+\]: delete `.+' from (shadow )?group `.+'$/ ) {
       # Ignore for now
    } elsif ( $ThisLine =~ /^xinetd\[\d+\]: USERID: ([^ ]+) (.+)$/ ) {
--- ./scripts/services/samba.org	Mon Jan 13 05:28:06 2003
+++ ./scripts/services/samba	Fri Mar 21 10:26:58 2003
@@ -40,7 +40,8 @@
          ($ThisLine =~ /Connection timed out/) or
          ($ThisLine =~ /closed connection to/) or
          ($ThisLine =~ /Got SIGHUP/) or
-         ($ThisLine =~ /current master browser/)
+         ($ThisLine =~ /current master browser/) or
+	 ($ThisLine =~ /debug_message/)
       ) {
       # Don't care about these...
    } elsif ( ($Host, $Service, $User) = ( $ThisLine =~ /([^ ]+ \([^ ]+\)) connect to service ([^ ]+) as user ([^ ]+)/ ) ) {
