Index: scripts/services/secure
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/secure,v
retrieving revision 1.46
diff -u -r1.46 secure
--- scripts/services/secure	21 Jun 2004 14:59:05 -0000	1.46
+++ scripts/services/secure	2 Oct 2004 12:47:45 -0000
@@ -52,16 +52,18 @@
    $temp =~ s/^([^[]+).*/$1/;
    if ($Ignore =~ /\b\Q$temp\E\b/i) { next; }
    if (
-      ( $ThisLine =~ m/^[^ ]+\[\d+\]: connect from localhost$/ ) or
+      ( $ThisLine =~ /^(.*)\(pam_unix\)/) or
+      ( $ThisLine =~ /^[^ ]+\[\d+\]: connect from localhost$/ ) or
       ( $ThisLine =~ /^\/usr\/bin\/sudo:/) or
-      ( $ThisLine =~ /^sudo:/) or
       ( $ThisLine =~ /^halt:/) or
-      ( $ThisLine =~ /^reboot:/) or
       ( $ThisLine =~ /^pam_xauth\[\d+\]: call_xauth: child returned \d/) or
       ( $ThisLine =~ /^passwd\[\d+\]:/) or
-      ( $ThisLine =~ /warning: can.t get client address: Connection refused/) or
+      ( $ThisLine =~ /^reboot:/) or
+      ( $ThisLine =~ /^sudo:/) or
       ( $ThisLine =~ /^xinetd\[\d+\]: USERID: ([^ ]+) (.+)$/ ) or
       ( $ThisLine =~ /^(xinetd|xinetd-ipv6)\[\d+\]: EXIT: /)
+      ( $ThisLine =~ /pam_succeed_if: requirement "uid < 100" not met by user /) or
+      ( $ThisLine =~ /warning: can.t get client address: Connection refused/) or
    ) {
       # Ignore these entries
    } elsif ($ThisLine =~ /^spop3d/) {
