Index: scripts/services/samba
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/samba,v
retrieving revision 1.17
diff -u -r1.17 samba
--- scripts/services/samba	15 Dec 2003 22:03:55 -0000	1.17
+++ scripts/services/samba	26 Jan 2004 15:18:33 -0000
@@ -2,6 +2,8 @@
 ##########################################################################
 # $Id$
 ##########################################################################
+# $Log$
+##########################################################################
 
 ########################################################
 # This was written and is maintained by:
@@ -13,46 +15,52 @@
 
 $Debug = $ENV{'LOGWATCH_DEBUG'};
 $Detail = $ENV{'LOGWATCH_DETAIL_LEVEL'} || 0;
-$SocketReadError=0;
-$DbOpenFail=0;
-$GetDomainMasterStatusFail=0;
+$SocketReadError = 0;
+$SocketWriteError = 0;
+$DbOpenFail = 0;
+$GetDomainMasterStatusFail = 0;
 
 if ( $Debug >= 5 ) {
-	print STDERR "\n\nDEBUG: Inside Samba Filter \n\n";
+   print STDERR "\n\nDEBUG: Inside Samba Filter \n\n";
 }
 
 while (defined($ThisLine = <STDIN>)) {
-	chomp($ThisLine);
-	if ( 
-		($ThisLine =~ /Currently not implemented/) or
-		($ThisLine =~ /version .+ started/) or
-		($ThisLine =~ /oplock[_ ]break/) or
-		($ThisLine =~ /No route to host/) or
-		($ThisLine =~ /SIGTERM: going down/) or
-		($ThisLine =~ /response packet id \d+ received with no matching record/) or
-		($ThisLine =~ /matchname/i) or
-		# Ignore entries in smbmount logfile
-		($ThisLine =~ /smbmount/) or
-		($ThisLine =~ /become_local_master/) or
-		($ThisLine =~ /become_domain_master/) or
-		($ThisLine =~ /add_domain_logon_names/) or
-		($ThisLine =~ /become_logon_server/) or
-		($ThisLine =~ /start_async_dns/) or
-		($ThisLine =~ /timeout connecting to/) or
-		($ThisLine =~ /Operation not permitted/) or
-		($ThisLine =~ /Record does not exist/) or
-		($ThisLine =~ /Connection reset by peer/) or
-		($ThisLine =~ /Multiple .+ responses received for a query/) or
-		($ThisLine =~ /Connection timed out/) or
-		($ThisLine =~ /closed connection to/) or
-		($ThisLine =~ /Got SIGHUP/) or
-		($ThisLine =~ /current master browser/) or
-		($ThisLine =~ /debug_message/) or
+   chomp($ThisLine);
+   if ( 
+      ($ThisLine =~ /Currently not implemented/) or
+      ($ThisLine =~ /version .+ started/) or
+      ($ThisLine =~ /oplock[_ ]break/) or
+      ($ThisLine =~ /No route to host/) or
+      ($ThisLine =~ /response packet id \d+ received with no matching record/) or
+      ($ThisLine =~ /matchname/i) or
+      # Ignore entries in smbmount logfile
+      ($ThisLine =~ /smbmount/) or
+      ($ThisLine =~ /become_local_master/) or
+      ($ThisLine =~ /become_domain_master/) or
+      ($ThisLine =~ /add_domain_logon_names/) or
+      ($ThisLine =~ /become_logon_server/) or
+      ($ThisLine =~ /start_async_dns/) or
+      ($ThisLine =~ /timeout connecting to/) or
+      ($ThisLine =~ /Operation not permitted/) or
+      ($ThisLine =~ /Record does not exist/) or
+      ($ThisLine =~ /Connection reset by peer/) or
+      ($ThisLine =~ /Multiple .+ responses received for a query/) or
+      ($ThisLine =~ /Connection timed out/) or
+      ($ThisLine =~ /closed connection to/) or
+      ($ThisLine =~ /current master browser/) or
+      ($ThisLine =~ /debug_message/) or
       ($ThisLine =~ /process_name_refresh_request\(184\)  Error - should be sent to WINS server$/) or
       ($ThisLine =~ /cli_connect\(783\)  Error connecting to [^ ]+ \(Operation already in progress\)$/) or
+      ($ThisLine =~ /nmbd_incomingrequests\.c:process_name_refresh_request\([0-9]+\)$/) or
+      ($ThisLine =~ /Got SIGHUP dumping debug info.$/) or
+      ($ThisLine =~ /Got SIGTERM: going down/) or
+      ($ThisLine =~ /get_socket_addr\(\d+\)  getpeername failed. Error was Transport endpoint is not connected$/) or
+      ($ThisLine =~ /lib\/access.c:check_access\(\d+\)$/) or
+      ($ThisLine =~ /smbd\/process.c:process_smb\(\d+\)$/) or
+      ($ThisLine =~ /nmbd\/nmbd_incomingdgrams.c:process_local_master_announce\(\d+\)$/) or
 		($ThisLine =~ /===============================================================/)
-	) {
-		#Don't care about these...
+   ) {
+      #Don't care about these...
 	} elsif ( ($Host, $Service, $User) = ( $ThisLine =~ /([^ ]+ \([^ ]+\)) connect to service ([^ ]+) as user ([^ ]+)/ ) ) {
 		$Connect{$Service}{$User}{$Host}++;
 	} elsif ( ($NoService) = ( $ThisLine =~ /couldn't find service (\S+)/ ) ) {
@@ -65,7 +73,7 @@
       $Temp = "$Where  ($Ip)";
       $Denied{$Temp}++;
       $CantGetGroup{$Browser}++;
-   } elsif ( ($Where,$Ip,$Name,$Group,$Subnet) = ($ThisLine =~ /(.*)  Denied connection from  \(([^ ]+)\)  Samba name server ([^ ]+) is now a local master browser for workgroup ([^ ]+) on subnet ([^ ]+)/ ) ) {
+   } elsif ( ($Where,$Ip,$Name,$Group,$Subnet) = ($ThisLine =~ /(.*)  Denied connection from  \(([^ ]+)\) [ *]+Samba name server ([^ ]+) is now a local master browser for workgroup ([^ ]+) on subnet ([^ ]+)/ ) ) {
       $Temp = "$Where  ($Ip)";
       $Denied{$Temp}++;
       $BeLocalMaster{$Subnet}{$Group}{$Name}++;
@@ -75,9 +83,16 @@
 		$NotFoundUser{$User}++;
 	} elsif (($User) = $ThisLine =~ /Rejecting user '([^ ]+)'/) {
 		$RejectedUser{$User}++;
-	} elsif ( ( $ThisLine =~ /^\[[^\]]+\] lib\/util_sock.c:read_data\(436\)/ ) ) {
+	} elsif ( ( $ThisLine =~ /lib\/util_sock.c:read_data\(436\)/ ) ) {
 		# This is due to a nasty bug in samba which causes it to drop connections :-(
 		$SocketReadError++;
+   } elsif (
+      ( $ThisLine =~ /lib\/util_sock.c:write_socket\(\d+\)  write_socket: Error writing \d bytes to socket/ ) or
+      ( $ThisLine =~ /lib\/util_sock.c:write_socket_data\(\d+\)  write_socket_data: write failure./ ) or 
+      ( $ThisLine =~ /lib\/util_sock.c:send_smb\(\d+\)  Error writing \d bytes to client. / )
+   ) {
+      # Something more generic should be here
+      $SocketWriteError++;
 	} elsif ( ( $ThisLine =~ /unable to open passdb database.$/ ) ) {
 		$DbOpenFail++;
 	} elsif ( ($Server,$Ip,$Group) = ($ThisLine =~ /Server ([^ ]+) at IP ([^ ]+) is announcing itself as a local master browser for workgroup ([^ ]+) and we think we are master. Forcing election.$/ ) ) {
@@ -107,15 +122,40 @@
 		$RootLoggedIn{$user}++;
 	} elsif ( ($file,$function) = ($ThisLine =~ /([a-zA-Z_\/():\.0-9-]+)  ([a-zA-Z0-9_-]+): Not yet implemented.$/)) {
 		$NotImplemented{$file}{$function}++;
+   } elsif ( ($User,$Ip,$Directory,$Reason) = ($ThisLine =~ /service.c:make_connection\([0-9]+\)  ([^ ]+) \(([^ ]+)\) Can't change directory to ([^ ]+) \((.*)\)/)) {
+      $Temp = "Netbios name $User on $Ip";
+      $CantChangeDir{$Directory}{$Reason}{$Temp}++;
+   } elsif ( ($Signal) = ($ThisLine =~ /open_sockets\([0-9]+\)  Reloading services after ([^ ]+)/)) {
+      $ReloadAfter{$Signal}++;
+   } elsif ( ($Signal) = ($ThisLine =~ /open_sockets\([0-9]+\)  Got ([^ ]+)/)) {
+      $ReloadAfter{$Signal}++;
+   } elsif ( ($Share,$Reason) = ($ThisLine =~ /cups_printername_ok\([0-9]+\)  (Unable to get printer status for [^ ]+) - ([^ ]+)/)) {
+      $PrinterStatus{$Share}{$Reason}++;
+   } elsif ( ($Share,$Reason) = ($ThisLine =~ /cups_queue_get\([0-9]+\)  (Unable to get jobs for [^ ]+) - ([^ ]+)/)) {
+      $PrinterStatus{$Share}{$Reason}++;
+   } elsif ( $ThisLine =~ m/main\([0-9]+\)  ERROR: Failed when creating subnet lists. Exiting./) {
+      $SubnetFail{"Failed when creating subnet lists"}++;
+   } elsif ( $ThisLine =~ m/create_subnets\([0-9]+\)  create_subnets: No local interfaces !/) {
+      $SubnetFail{"No local interfaces"}++;
+   } elsif ( $ThisLine =~ m/reload_interfaces: No subnets to listen to. Shutting down.../) {
+      $SubnetFail{"No subnets to listen to. Shutting down."}++;
+   } elsif ( $ThisLine =~ s/process_get_backup_list_request\([0-9]+\)  process_get_backup_list_request: (.*)/$1/) {
+      $GetBacupList{$ThisLine}++;
+   } elsif ( ($Error) = ($ThisLine =~ /brl_init\([0-9]+\)  (Failed to open byte range locking database)$/)) {
+      $LockDbError{$Error}++;
+   } elsif ( ($Error) = ($ThisLine =~ /locking_init\([0-9]+\)  ERROR: (Failed to initialise locking database)$/)) {
+      $LockDbError{$Error}++;
+   } elsif ( ($Location,$Reason) = ($ThisLine =~ /tdb_log\([0-9]+\)  tdb\(([^ ]+)\): tdb_reopen: (open failed \([^ ]+\))/)) {
+      $LockDbError{"$Location - $Reason"}++;
 	} else {
 		# Report any unmatched entries...
 		$OtherList{$ThisLine}++;
 
       #TODO:
-      #lib/util_sock.c:get_socket_addr(1012)  getpeername failed. Error was Transport endpoint is not connected
-      #lib/util_sock.c:send_smb(704)  Error writing 4 bytes to client. -1. (Broken pipe)
-      #lib/util_sock.c:write_socket(524)  write_socket: Error writing 4 bytes to socket 5: ERRNO = Broken pipe
-      #lib/util_sock.c:write_socket_data(499)  write_socket_data: write failure. Error = Broken pipe
+      #smbd/oplock.c:process_local_message(418)  process_local_message: unknown UDP message command code (424d) - ignoring.
+      #smbd/process.c:switch_message(662)  Non-SMB packet of length 156. Terminating server
+      #smbd/process.c:switch_message(662)  Non-SMB packet of length 133. Terminating server
+      #libsmb/nmblib.c:send_udp(756)  Packet send failed to 153.19.207.127(138) ERRNO=Invalid argument
       #lib/util_sock.c:read_data(436)  read_data: read failure for 4. Error = Brak drogi do systemu
 	}
 }
@@ -131,6 +171,21 @@
 	}
 }
 
+if (keys %SubnetFail) {
+   print "\nWARNING!!!!!!\n";
+   print "Errors when creating subnets:\n";
+   foreach $Error (sort {$a cmp $b} keys %SubnetFail) {
+      print "   $Error : $SubnetFail{$Error} Time(s)\n";
+   }
+}
+
+if (keys %ReloadAfter) {
+   print "\nReloaded services after signal:\n";
+   foreach $Signal (sort {$a cmp $b} keys %ReloadAfter) {
+      print "   $Signal : $ReloadAfter{$Signal} Time(s)\n";
+   }
+}
+
 if (keys %DiscardWins) {
    print "\nDiscarded invalid wins.dat file with version:\n";
    foreach $Version (sort {$a cmp $b} keys %DiscardWins) {
@@ -169,6 +224,16 @@
    }
 }
 
+if (keys %PrinterStatus) {
+   print "\nPrinter Errors:\n";
+   foreach $Share (sort {$a cmp $b} keys %PrinterStatus) {
+      print "   $Share:\n";
+      foreach $Reason (sort {$a cmp $b} keys %{$PrinterStatus{$Share}}) {
+         print "      $Reason : $PrinterStatus{$Share}{$Reason} Time(s)\n";
+      }
+   }
+}
+
 if (($Detail >= 5) and (keys %RootLoggedIn)) {
    print "\nAdmin logins (root privileges):\n";
    foreach $user (sort {$a cmp $b} keys %RootLoggedIn) {
@@ -186,13 +251,6 @@
 }
      
 
-if (keys %Denied) {
-   print "\nConnections Denied:\n";
-   foreach $Line (sort {$a cmp $b} keys %Denied) {
-      print "   $Line : $Denied{$Line} Time(s)\n";
-   }
-}
-
 if (keys %ForceElection) {
 	print "\nForced Election:\n";
 	foreach $Group (sort {$a cmp $b} keys %ForceElection) {
@@ -227,6 +285,13 @@
 	print "\nFailed to get Domain Master node name: $GetDomainMasterStatusFail Time(s)\n";
 }
 
+if (keys %GetBacupList) {
+   print "\nBackup list requests:\n";
+   foreach $Request (sort {$a cmp $b} keys %GetBacupList) {
+      print "   $Request : $GetBacupList{$Request} Time(s)\n";
+   }
+}
+
 if (($Detail >= 5) and (keys %NoServ)) {
    print "\nCouldn't find services:\n";
    foreach $ThisOne (sort {$a cmp $b} keys %NoServ) {
@@ -293,8 +358,28 @@
     }
 }
 
+if (keys %CantChangeDir) {
+   print "\nCan't change directory while browsing:\n";
+   foreach $Directory (sort {$a cmp $b} keys %CantChangeDir) {
+      print "   $Directory:\n";
+      foreach $Reason (sort {$a cmp $b} keys %{$CantChangeDir{$Directory}}) {
+         print "      $Reason:\n";
+         foreach $Entry (sort {$a cmp $b} keys %{$CantChangeDir{$Directory}{$Reason}}) {
+            print "         $Entry : $CantChangeDir{$Directory}{$Reason}{$Entry} Time(s)\n";
+         }
+      }
+   }
+}
+
 if ($SocketReadError > 0) {
 	print "\nSocket Read Error (Samba bug): $SocketReadError Time(s)\n";
+}
+
+if (keys %LockDbError) {
+   print "\nLocking Database error:\n";
+   foreach $Error (sort {$a cmp $b} keys %LockDbError) {
+      print "   $Error : $LockDbError{$Error} Time(s)\n";
+   }
 }
 
 if (keys %OtherList) {
