Index: scripts/logwatch.pl
===================================================================
RCS file: /var/cvs/logwatch/scripts/logwatch.pl,v
retrieving revision 1.119
diff -u -r1.119 logwatch.pl
--- scripts/logwatch.pl	24 Feb 2005 17:10:33 -0000	1.119
+++ scripts/logwatch.pl	24 Feb 2005 22:25:19 -0000
@@ -21,10 +21,11 @@
 ########################################################
 
 #To be FHS compliant distro packages might use this
-#my $BaseDir = "/usr/share/logwatch";
+my $BaseDir = "/usr/share/logwatch";
+my $ConfigDir = "/etc/log.d";
 #my $ConfigDir = "/etc/log.d/conf";
-my $BaseDir = "/etc/log.d";
-my $ConfigDir = "$BaseDir/conf";
+#my $BaseDir = "/etc/log.d";
+#my $ConfigDir = "$BaseDir/conf";
 my $PerlVersion = "/usr/bin/perl";
 
 #############################################################################
