Index: scripts/services/postfix
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/postfix,v
retrieving revision 1.13
diff -u -r1.13 postfix
--- scripts/services/postfix	23 Jun 2004 15:01:17 -0000	1.13
+++ scripts/services/postfix	30 Jun 2004 14:25:14 -0000
@@ -188,8 +188,10 @@
    } elsif ( ($Command,$Host) = ($ThisLine =~ /timeout after ([^ ]*) from ([^ ]*)$/)) {
       # Make some better summary with hosts
       $ConnectionLost{$Command}++;
-   } elsif ( ($Rejected,undef,undef) = ($ThisLine =~ /^[a-zA-Z0-9]+: reject: header (.*); from=<([^ ]*)> to=<([^ ]*)>: Message content rejected$/)) {
-      $HeaderReject{$Rejected}++;
+   } elsif ( ($Rejected,undef,undef,$Reason) = ($ThisLine =~ /^[a-zA-Z0-9]+: reject: header (.*); from=<([^ ]*)> to=<([^ ]*)>: (.*)$/)) {
+      $HeaderReject{$Reason}{$Rejected}++;
+   } elsif ( ($Warning,undef,undef,$Reason) = ($ThisLine =~ /^[a-zA-Z0-9]+: warning: header (.*); from=<([^ ]*)> to=<([^ ]*)>: (.*)$/)) {
+      $HeaderWarning{$Reason}{$Warning}++;
    #} elsif ( ($Host,undef) = ($ThisLine =~ /reject: RCPT from ([^ ]*\[[^ ]*\]): [0-9]+ <([^ ]*)>: Sender address rejected: Domain not found;/)) {
    #   $RejectDomain{$Host}++;
    # above two lines included in generic reject sender on next condition
@@ -462,8 +464,19 @@
 
 if (($Detail >= 5) and (keys %HeaderReject)) {
    print "\n\nHeader content reject:\n";
-   foreach $ThisOne (sort {$a cmp $b} keys %HeaderReject) {
-      print "   $ThisOne : $HeaderReject{$ThisOne} Time(s)\n";
+   foreach $Reason (sort {$a cmp $b} keys %HeaderReject) {
+      print "   $Reason:";
+      foreach $Rejected (sort {$a cmp $b} keys %{$HeaderReject{$Reason}} ) {
+         print "      $Rejected : $HeaderReject{$Reason}{$Rejected} Time(s)\n";
+   }
+}
+
+if (($Detail >= 5) and (keys %HeaderWarning)) {
+   print "\n\nHeader content reject:\n";
+   foreach $Reason (sort {$a cmp $b} keys %HeaderWarning) {
+      print "   $Reason:";
+      foreach $Warning (sort {$a cmp $b} keys %{$HeaderWarning{$Reason}} ) {
+         print "      $Warning : $HeaderWarning{$Reason}{$Warning} Time(s)\n";
    }
 }
 
