diff -u 5.2.2/samba new/samba
--- 5.2.2/samba	2004-06-25 01:22:09.000000000 -0700
+++ new/samba	2004-08-30 06:46:03.123311000 -0700
@@ -34,6 +34,9 @@
 while (defined($ThisLine = <STDIN>)) {
    chomp($ThisLine);
    if ( 
+      ($ThisLine =~ /smbd\/service.c:make_connection_snum\(\d+\)  .+ connect to service .+ initially as user/) or
+      ($ThisLine =~ /smbd\/server.c:open_sockets_smbd\(\d+\)  Reloading services after SIGHUP/) or
+      ($ThisLine =~ /lib\/util_sock.c:get_peer_addr\(\d+\)  getpeername failed. Error was Transport endpoint is not connected/) or
       ($ThisLine =~ /add_domain_logon_names/) or
       ($ThisLine =~ /become_domain_master/) or
       ($ThisLine =~ /become_local_master/) or
diff -u 5.2.2/secure new/secure
--- 5.2.2/secure	2004-06-25 01:22:09.000000000 -0700
+++ new/secure	2004-08-30 06:51:52.126770000 -0700
@@ -52,6 +52,7 @@
    $temp =~ s/^([^[]+).*/$1/;
    if ($Ignore =~ /\b\Q$temp\E\b/i) { next; }
    if (
+      ( $ThisLine =~ /pam_succeed_if: requirement "uid < 100" not met by user /) or
       ( $ThisLine =~ m/^[^ ]+\[\d+\]: connect from localhost$/ ) or
       ( $ThisLine =~ /^\/usr\/bin\/sudo:/) or
       ( $ThisLine =~ /^sudo:/) or
diff -u 5.2.2/sshd new/sshd
--- 5.2.2/sshd	2004-08-30 06:37:05.089330000 -0700
+++ new/sshd	2004-08-30 06:51:41.541758000 -0700
@@ -59,6 +59,7 @@
    }
    chomp($ThisLine);
    if (
+       ($ThisLine =~ /^pam_succeed_if: requirement "uid < 100" not met by user /) or
        ($ThisLine =~ m/^(log: )?$/ ) or
        ($ThisLine =~ m/^(log: )?\^\[\[60G/ ) or
        ($ThisLine =~ m/^(log: )? succeeded$/ ) or
