Index: scripts/services/postfix
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/postfix,v
retrieving revision 1.9
diff -u -r1.9 postfix
--- scripts/services/postfix	3 Feb 2004 03:25:02 -0000	1.9
+++ scripts/services/postfix	11 Feb 2004 09:56:01 -0000
@@ -155,7 +155,9 @@
    #   $RejectDomain{$Host}++;
    # above two lines included in generic reject sender on next condition
    } elsif ( ($Host,$Sender,$Reason) = ($ThisLine =~ /reject: RCPT from ([^ ]*\[[^ ]*\]): [0-9]+ <([^ ]*)>: Sender address rejected: (.*);/)) {
-      $RejectSender{$Host}++;
+      $RejectSender{$Reason}{$Host}{$Sender}++;
+      $RejectSenderHost{$Reason}{$Host}++;
+      $RejectSenderReason{$Reason}++;
    } elsif ( ($Host,$Recip,$Reason) = ($ThisLine =~ /reject: RCPT from ([^ ]*\[[^ ]*\]): [0-9]+ <([^ ]*)>: Recipient address rejected: (.*);/)) {
       $Temp = "$Host : $Reason";
       $RejectRecip{$Recip}{$Temp}++;
@@ -362,11 +364,16 @@
 #}
 
 if (keys %RejectSender) {
-   print "\n\nMessages rejected from sender:\n";
-   foreach $Sender (sort {$a cmp $b} keys %RejectSender) {
-      print "    $Sender:\n";
-      foreach $Host (sort {$a cmp $b} keys %{$RejectSender{$Sender}} ) {
-         print "        $Host : $RejectSender{$Sender}{$Host} Times(s)\n";
+   print "\n\nMessages rejected:\n";
+   foreach $Reason (sort {$a cmp $b} keys %RejectSender) {
+      print "    $Reason $RejectSenderReason{$Reason} Time(s)\n";
+      foreach $Host (sort {$a cmp $b} keys %{$RejectSender{$Reason}} ) {
+         print "        $Host $RejectSenderHost{$Reason}{$Host} Time(s)\n";
+         if ($Detail >= 5) {
+            foreach $Sender (sort {$a cmp $b} keys %{$RejectSender{$Reason}{$Host}} ) {
+               print "           $Sender : $RejectSender{$Reason}{$Host}{$Sender} Time(s)\n";
+            }
+         }
       }
    }
 }
@@ -376,7 +383,7 @@
    foreach $Recip (sort {$a cmp $b} keys %RejectRecip) {
       print "    $Recip:\n";
       foreach $Host (sort {$a cmp $b} keys %{$RejectRecip{$Recip}} ) {
-         print "        $Host : $RejectRecip{$Recip}{$Host} Times(s)\n";
+         print "        $Host : $RejectRecip{$Recip}{$Host} Time(s)\n";
       }
    }
 }
@@ -508,7 +515,7 @@
 if (keys %UnknownWarnings) {
    print "\n\nUnrecognized warning:\n";
    foreach $ThisOne (sort {$a cmp $b} keys %UnknownWarnings) {
-      print "    $ThisOne : $UnknownWarnings{$ThisOne} Times(s)\n";
+      print "    $ThisOne : $UnknownWarnings{$ThisOne} Time(s)\n";
    }
 }
 
