--- ./scripts/services/pam_unix.org	Sat Oct 12 04:08:19 2002
+++ ./scripts/services/pam_unix	Thu Mar 20 09:12:05 2003
@@ -60,6 +60,30 @@
       } else {
          $data{$service}{'Unknown Entries'}{$line}++;
       }
+   } elsif ($service eq 'imap') {
+	   if ($line =~ s/^authentication failure; logname=(.+) uid=(\d+) .*user=(.+)$/$1($2) -> $3/) {
+		   $data{$service}{'Authentication Failures'}{$line}++;
+	   } else {
+		   $data{$service}{'Unknown Entries'}{$line}++;
+	   }
+   } elsif ($service eq 'spop3d') {
+	   if ($line =~ s/^session opened for user (.+)/$1/) {
+		   $data{$service}{'Sessions Opened'}{$line}++;
+	   } elsif ($line =~ /session closed for user/) {
+		   # ignore this line
+	   } elsif ($line =~ s/^authentication failure; logname=(.+) uid=(\d+) .*user=(.+)$/$1($2) -> $3/) {
+		   $data{$service}{'Authentication Failures'}{$line}++;
+	   } else {
+		   $data{$service}{'Unknown Entries'}{$line}++;
+	   }
+   } elsif ($service eq 'cron') {
+	   if ($line =~ s/^session opened for user (.+) by \(uid=\d+\)/$1/) {
+		   ($Detail >= 5) && $data{$service}{'Sessions Opened'}{$line}++;
+	   } elsif ($line =~ /session closed for user/) {
+		   # ignore this line
+	   } else {
+		   $data{$service}{'Unknown Entries'}{$line}++;
+	   }
    } else {
       $data{$service}{'Unknown Entries'}{$line}++;
    }
