Index: scripts/services/samba
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/samba,v
retrieving revision 1.12
diff -u -r1.12 samba
--- scripts/services/samba	26 Nov 2003 14:36:30 -0000	1.12
+++ scripts/services/samba	7 Dec 2003 17:26:10 -0000
@@ -57,6 +57,8 @@
 		$NoServ{$NoService}++;
 	} elsif ($ThisLine =~ s/Denied connection from\s+(\S+)$/$1/) {
 		$Denied{$ThisLine}++;
+	} elsif ($ThisLine =~ s/ Connection denied from\s+(\S+)$/$1/) {
+		$Denied{$ThisLine}++;
 	} elsif ( ($Ip,$Name,$Group,$Subnet) = ($ThisLine =~ /Denied connection from  \(([^ ]+)\)  Samba name server ([^ ]+) is now a local master browser for workgroup ([^ ]+) on subnet ([^ ]+)/ ) ) {
 		$Denied{$Ip}++;
 		$BeLocalMaster{$Subnet}{$Group}{$Name}++;
@@ -88,9 +90,17 @@
 		$Crash{$Temp}++;
 	} elsif ( ( $ThisLine =~ /get_domain_master_name_node_status_fail\(([^ ]+)\)/ ) ) {
 		$GetDomainMasterStatusFail++;
+	} elsif ( ($User) = ($ThisLine =~ /pass_check_smb\(552\)  Account for user '([^ ]+)' was disabled.$/) ) {
+		$AccountDisabled{$User}++;
 	} else {
 		# Report any unmatched entries...
 		$OtherList{$ThisLine}++;
+		#
+		#TODO:
+		#lib/util_sock.c:get_socket_addr(1012)  getpeername failed. Error was Transport endpoint is not connected
+		#lib/util_sock.c:send_smb(704)  Error writing 4 bytes to client. -1. (Broken pipe)
+		#lib/util_sock.c:write_socket(524)  write_socket: Error writing 4 bytes to socket 5: ERRNO = Broken pipe
+		#lib/util_sock.c:write_socket_data(499)  write_socket_data: write failure. Error = Broken pipe
 	}
 }
 
@@ -120,7 +130,7 @@
 
 if (keys %Denied) {
    print "\nConnections Denied:\n";
-   foreach $Line (keys %Denied) {
+   foreach $Line (sort {$a cmp $b} keys %Denied) {
       print "   $Line : $Denied{$Line} Time(s)\n";
    }
 }
@@ -199,23 +209,30 @@
 
 if (keys %InvalidUser) {
    print "\nInvalid Users:\n";
-   foreach $Line (keys %InvalidUser) {
+   foreach $Line (sort {$a cmp $b} keys %InvalidUser) {
       print "   $Line : $InvalidUser{$Line} Time(s)\n";
    }
 }
 
 if (keys %NotFoundUser) {
    print "\nUsers not found in UNIX Database:\n";
-   foreach $Line (keys %NotFoundUser) {
+   foreach $Line (sort {$a cmp $b} keys %NotFoundUser) {
       print "   $Line : $NotFoundUser{$Line} Time(s)\n";
    }
 }
 
 if (keys %RejectedUser) {
    print "\nRejected Users:\n";
-   foreach $Line (keys %RejectedUser) {
+   foreach $Line (sort {$a cmp $b} keys %RejectedUser) {
       print "   $Line : $RejectedUser{$Line} Time(s)\n";
    }
+}
+
+if (keys %AccountDisabled) {
+	print "\nAccounts disabled:\n";
+	foreach $User (sort {$a cmp $b} keys %AccountDisabled) {
+		print "   $User : $AccountDisabled{$User} Time(s)\n";
+	}
 }
 
 if ($SocketReadError > 0) {
