Index: scripts/services/postfix
===================================================================
RCS file: /var/cvs/logwatch/scripts/services/postfix,v
retrieving revision 1.4
diff -u -r1.4 postfix
--- scripts/services/postfix	26 Nov 2003 14:36:30 -0000	1.4
+++ scripts/services/postfix	7 Dec 2003 17:46:50 -0000
@@ -55,6 +55,8 @@
 $BytesTransferred = 0;
 $FourHourReturns = 0;
 $ReturnedToSender = 0;
+$ResentMessages = 0;
+$RemovedFromQueue = 0;
 
 while (defined($ThisLine = <STDIN>)) {
 	if ( ( $ThisLine =~ m/^connect/ ) or 
@@ -69,19 +71,18 @@
 		( $ThisLine =~ m/^warning: no MX host for .* has a valid A record$/ ) or
 		( $ThisLine =~ m/^warning: numeric domain name in resource data of MX record for .*$/ ) or
 		( $ThisLine =~ m/^daemon started$/ ) or
-		( $ThisLine =~ m/^terminating on signal 15$/ )
+		( $ThisLine =~ m/^terminating on signal 15$/ ) or
+		( $ThisLine =~ m/^warning: Mail system is down -- accessing queue directly$/ ) or
+		( $ThisLine =~ m/^Deleted: \d message$/ )
 	) {
 		# We don't care about these
-	}
-	elsif ( ($Bytes) = ($ThisLine =~ /^[a-zA-Z0-9]+: from=.*size=([0-9]+).*$/) ) {
+	} elsif ( ($Bytes) = ($ThisLine =~ /^[a-zA-Z0-9]+: from=.*size=([0-9]+).*$/) ) {
 		$MsgsSent++;
 		$BytesTransferred += $Bytes;
-    }
-    elsif (($User) = ($ThisLine =~ /^[a-zA-Z0-9]+: to\=\<([^ ]*)>, relay\=local, delay\=[0-9]+, status\=bounced \(unknown user/)) {
+    } elsif (($User) = ($ThisLine =~ /^[a-zA-Z0-9]+: to\=\<([^ ]*)>, relay\=local, delay\=[0-9]+, status\=bounced \(unknown user/)) {
 		# unknown user
 		$UnknownUsers{$User}++;
-    }
-    elsif (($Dest, $Relay, $Msg) = ($ThisLine =~ /^[a-zA-Z0-9]+: to\=\<([^ ]*)>, relay=([^ ]*).*, delay\=[0-9]+, status\=bounced \(([^)]*)/ )) {
+    } elsif (($Dest, $Relay, $Msg) = ($ThisLine =~ /^[a-zA-Z0-9]+: to\=\<([^ ]*)>, relay=([^ ]*).*, delay\=[0-9]+, status\=bounced \(([^)]*)/ )) {
 		# unknown user
 		# $Msg = " hello "
 		# print "bounce message from " . $Dest . " msg : " . $Relay . "\n";
@@ -92,54 +93,65 @@
 			$Temp = "To " . $Dest . " Msg=\"" . $Msg . "\"";
 			$ForeignBounce{$Temp}++;
 		}
-	}
-	elsif ( ($Relay,$Dest) = ($ThisLine =~ m/^reject: RCPT from ([^ ]*): 554 <([^ ]*)>.* Relay access denied.* to=([^ ]*)/) ) {
+	} elsif ( ($Relay,$Dest) = ($ThisLine =~ m/^reject: RCPT from ([^ ]*): 554 <([^ ]*)>.* Relay access denied.* to=([^ ]*)/) ) {
 		# print "reject: " . $ThisLine . "\n";
 		# print "Relay :" . $Relay . " to " . $Dest . "\n";
 		$Temp = "From " . $Relay . " to " . $Dest;
 		$RelayDenied{$Temp}++;
-    }
-    elsif ( ($User,$From) = ($ThisLine =~ /^[a-zA-Z0-9]+: uid=([^ ]*) from=\<([^ ]*)>/)) {
+    } elsif ( ($User,$From) = ($ThisLine =~ /^[a-zA-Z0-9]+: uid=([^ ]*) from=\<([^ ]*)>/)) {
 		#Messages sent by user
 		$Temp = $From . " (uid=" . $User . "): ";
 		$SentBy{$Temp}++;
-	}
-	elsif ( ($From) = ($ThisLine =~ /^[a-zA-Z0-9]+: from=<([^ ]*)>, status=expired, returned to sender$/)) {
+	} elsif ( ($From) = ($ThisLine =~ /^[a-zA-Z0-9]+: from=<([^ ]*)>, status=expired, returned to sender$/)) {
 		$ReturnedToSender++;
-	}
-	elsif ( ($Command,$Host) = ($ThisLine =~ /^lost connection after ([^ ]*) from ([^ ]*)$/)) {
+	} elsif ( (undef) = ($ThisLine =~ /^[a-zA-Z0-9]+: resent-message-id=<([^ ]*)>$/)) {
+		$ResentMessages++;
+	} elsif ( ($Command,$Host) = ($ThisLine =~ /^lost connection after ([^ ]*) from ([^ ]*)$/)) {
 		# Make some better summary with hosts
 		$ConnectionLost{$Command}++;
-	}
-	elsif ( ($Command,$Host) = ($ThisLine =~ /^timeout after ([^ ]*) from ([^ ]*)$/)) {
+	} elsif ( ($Command,$Host) = ($ThisLine =~ /^timeout after ([^ ]*) from ([^ ]*)$/)) {
 		# Make some better summary with hosts
 		$ConnectionLost{$Command}++;
-	}
-	elsif ( ($Rejected,undef,undef) = ($ThisLine =~ /^^[a-zA-Z0-9]+: reject: header (.*); from=<([^ ]*)> to=<([^ ]*)>: Message content rejected$/)) {
+	} elsif ( ($Rejected,undef,undef) = ($ThisLine =~ /^[a-zA-Z0-9]+: reject: header (.*); from=<([^ ]*)> to=<([^ ]*)>: Message content rejected$/)) {
 		$HeaderReject{$Rejected}++;
-	}
-#    elsif ( ($Warning)  = ($ThisLine =~ /^[a-zA-Z0-9]+: Authentication-Warning: [^ ]+: ([^ ]+ set sender to [^ ]+ using -f)/) ) {
-#        $AuthWarns{$Warning}++;
-#    }
-#    elsif ( ($Forward,$Error) = ($ThisLine =~ /^[a-zA-Z0-9]+: forward ([^ ]*): transient error: (.*)$/) ) {
-#        $Temp = $Forward . ": " . $Error;
-#        $ForwardErrors{$Temp}++;
-#    }
-#    elsif ( ($Temp) = ($ThisLine =~ /^[a-zA-Z0-9]+: forward ([^ ]*: Permission denied)/) ) {
-#        $ForwardErrors{$Temp}++;
-#    }
-#    elsif ( $ThisLine =~ m/^[a-zA-Z0-9]+: [a-zA-Z0-9]+: return to sender: Warning: could not send message for past 4 hours/ ) {
-#        $FourHourReturns++;
-#    }
-    else {
-	push @OtherList,$ThisLine;
+	} elsif ( ($Host,undef) = ($ThisLine =~ /^reject: RCPT from ([^ ]*\[[^ ]*\]): 450 <([^ ]*)>: Sender address rejected: Domain not found;/)) {
+		$RejectDomain{$Host}++;
+	} elsif ( ($Host,$Site,$Reason) = ($ThisLine =~ /^reject: RCPT from ([^ ]*\[[^ ]*\]): 554 Service unavailable; \[[^ ]*\] blocked using ([^ ]*), reason: (.*);/)) {
+		$Temp = "$Host : $Reason";
+		$RejectRBL{$Site}{$Temp}++;
+	} elsif ( (undef,undef,$Error) = ($ThisLine =~ /^warning: ([^ ]*): hostname ([^ ]*) verification failed: ([^ ]*)$/)) {
+		$HostnameVerification{$Error}++;
+	} elsif ( $ThisLine =~ /^[a-zA-Z0-9]+: removed$/) {
+		$RemovedFromQueue++;
+	} elsif ( ($Host) = ($ThisLine =~ /^[a-zA-Z0-9]+: enabling PIX <CRLF>.<CRLF> workaround for ([^ ]*\[[^ ]*\])$/)) {
+		$PixWorkaround{$Host}++;
+	} elsif ( ($Message) = ($ThisLine =~ /^warning: valid_hostname: (.*)$/)) {
+		$ValidHostname{$Message}++;
+	} else {
+		push @OtherList,$ThisLine;
+		#
+		# TODO:
+		#warning: Illegal address syntax from f133.ds.univ.gda.pl[153.19.197.133] in MAIL command: <sseeds_pg@teamaffiliate.com onMouseOver=>
+		#warning: bad size limit "(5120000)" in EHLO reply from relay.hwk-aachen.de[212.117.75.130]
+		#warning: host 127.0.1.50[127.0.1.50] greeted me with my own hostname jol.task.gda.pl
+		#warning: host 127.0.1.50[127.0.1.50] replied to HELO/EHLO with my own hostname jol.task.gda.pl
+		#warning: mailer loop: best MX host for pager.icq.com is local
+		#warning: malformed domain name in resource data of MX record for sfol.com:
+		#warning: malformed domain name in resource data of MX record for kiri.com: \254\184\194\008
+		#warning: open active EA67A1B840: No such file or directory
+		#warning: premature end-of-input from cleanup socket while reading input attribute name
+		#warning: smtp_connect_addr: socket: Address family not supported by protocol
+		#warning: smtp_addr_one: unknown address family 1 for localhost
+		#warning: unknown[203.197.217.137] sent message header instead of SMTP command: From:"Bailey" <sanchez_24@canada.com>
     }
 }
 
 if ((@OtherList) or
 	($MsgsSent > 0) or
 	($FourHourReturns > 0) or
-	($ReturnedToSender >0) or
+	($ReturnedToSender > 0) or
+	($ResentMessages > 0) or
+	($$RemovedFromQueue > 0) or
 	(keys %UnknownUsers) or
 	(keys %RelayDenied) or
 	(keys %$HeaderReject) or
@@ -148,83 +160,132 @@
 	(keys %AuthWarns) or
 	(keys %ForwardErrors) or
 	(keys %SentBy) or
-	(keys %ConnectionLost)
+	(keys %ConnectionLost) or
+	(keys %HostnameVerification) or
+	(keys %PixWorkaround) or
+	(keys %$ValidHostname)
 ) {
 	print "\n\n --------------------- postfix Begin ------------------------ ";
 	
     if($MsgsSent > 0) {
-        print "\n\n" . $BytesTransferred . " bytes transferred";
-        print "\n" . $MsgsSent . " messages sent";
+        print "\n\n$BytesTransferred bytes transferred";
+        print "\n$MsgsSent messages sent";
     }
 
-    if($FourHourReturns > 0) {
-        print "\n\n" . $FourHourReturns . " messages returned after 4 hours";
+    if ($FourHourReturns > 0) {
+        print "\n\n$FourHourReturns messages returned after 4 hours";
     }
 
 	if ($ReturnedToSender >0) {
-		print "\n\n" . $ReturnedToSender . " messages expired and returned to sender";
+		print "\n\n$ReturnedToSender messages expired and returned to sender";
+	}
+
+	if ($ResentMessages > 0) {
+		print "\n\n$ResentMessages resent messages";
+	}
+
+	if ($RemovedFromQueue > 0) {
+		print "\n\n$RemovedFromQueue messages removed from queue";
+	}
+
+	if (keys %PixWorkaround) {
+		print "\n\nEnabled PIX <CRLF>.<CRLF> workaround for:\n";
+		foreach $Host (sort {$a cmp $b} keys %PixWorkaround) {
+			print "    $Host : $PixWorkaround{$Host}  Times(s)\n";
+		}
 	}
 
 	if (keys %SentBy) {
 		print "\n\nMessages sent by:\n";
-		foreach $ThisOne (keys %SentBy) {
-			print "    " . $ThisOne . $SentBy{$ThisOne} . " Times(s)\n";
+		foreach $ThisOne (sort {$a cmp $b} keys %SentBy) {
+			print "    $ThisOne : $SentBy{$ThisOne} Times(s)\n";
 		}
 	}
 
     if (keys %UnknownUsers) {
 		print "\n\nUnknown users:\n";
-		foreach $ThisOne (keys %UnknownUsers) {
-			print "    " . $ThisOne . ": " . $UnknownUsers{$ThisOne} . " Times(s)\n";
+		foreach $ThisOne (sort {$a cmp $b} keys %UnknownUsers) {
+			print "    $ThisOne : $UnknownUsers{$ThisOne} Times(s)\n";
 		}
 	}
 
     if (keys %RelayDenied) {
 		print "\n\nRelaying denied:\n";
-		foreach $ThisOne (keys %RelayDenied) {
+		foreach $ThisOne (sort {$a cmp $b} keys %RelayDenied) {
 			print "    " . $ThisOne . ": " . $RelayDenied{$ThisOne} . " Times(s)\n";
 		}
     }
 
     if (keys %LocalBounce) {
 		print "\n\nLocal Bounce:\n";
-		foreach $ThisOne (keys %LocalBounce) {
+		foreach $ThisOne (sort {$a cmp $b} keys %LocalBounce) {
 			print "    " . $ThisOne . ": " . $LocalBounce{$ThisOne} . " Times(s)\n";
 		}
     }
 
     if (keys %ForeignBounce) {
 		print "\n\nForeign Bounce:\n";
-		foreach $ThisOne (keys %ForeignBounce) {
+		foreach $ThisOne (sort {$a cmp $b} keys %ForeignBounce) {
 			print "    " . $ThisOne . ": " . $ForeignBounce{$ThisOne} . " Times(s)\n";
 		}
     }
 
 	if (keys %HeaderReject) {
 		print "\n\nHeader content reject:\n";
-		foreach $ThisOne (keys %HeaderReject) {
+		foreach $ThisOne (sort {$a cmp $b} keys %HeaderReject) {
 			print "    " . $ThisOne . ": " . $HeaderReject{$ThisOne} . " Times(s)\n";
 		}
 	}
 
+	if (keys %RejectDomain) {
+		print "\n\nNot found domain in address sent by:\n";
+		foreach $Host (sort {$a cmp $b} keys %RejectDomain) {
+			print "    $Host : $RejectDomain{$Host} Times(s)\n";
+		}
+	}
+
+	if (keys %RejectRBL) {
+		print "\n\nMessages rejected using Anti-Spam site:\n";
+		foreach $Site (sort {$a cmp $b} keys %RejectRBL) {
+			print "    $Site:\n";
+			foreach $Host (sort {$a cmp $b} keys %{$RejectRBL{$Site}} ) {
+				print "        $Host : $RejectRBL{$Site}{$Host} Times(s)\n";
+			}
+		}
+	}
+
     if (keys %AuthWarns) {
 		print "\n\nAuthentication warnings:\n";
-		foreach $ThisOne (keys %AuthWarns) {
-			print "    " . $ThisOne . ": " . $AuthWarns{$ThisOne} . " Times(s)\n";
+		foreach $ThisOne (sort {$a cmp $b} keys %AuthWarns) {
+			print "    $ThisOne : $AuthWarns{$ThisOne} Times(s)\n";
 		}
     }
 
     if (keys %ForwardErrors) {
 		print "\n\nForwarding errors:\n";
-		foreach $ThisOne (keys %ForwardErrors) {
-			print "    " . $ThisOne . ": " . $ForwardErrors{$ThisOne} . " Times(s)\n";
+		foreach $ThisOne (sort {$a cmp $b} keys %ForwardErrors) {
+			print "    $ThisOne : $ForwardErrors{$ThisOne} Times(s)\n";
 		}
 	}
 
 	if (keys %ConnectionLost) {
 		print "\n\nConnections lost:\n";
-		foreach $ThisOne (keys %ConnectionLost) {
-			print "    Connection lost after command " . $ThisOne . ": " . $ConnectionLost{$ThisOne} . " Times(s)\n";
+		foreach $ThisOne (sort {$a cmp $b} keys %ConnectionLost) {
+			print "    Connection lost after command $ThisOne : $ConnectionLost{$ThisOne} Times(s)\n";
+		}
+	}
+
+	if (keys %HostnameVerification) {
+		print "\n\nHostname verification errors:\n";
+		foreach $Error (sort {$a cmp $b} keys %HostnameVerification) {
+			print "    $Error : $HostnameVerification{$Error} Times(s)\n";
+		}
+	}
+
+	if (keys %ValidHostname) {
+		print "\n\nHostname validation errors:\n";
+		foreach $Message (sort {$a cmp $b} keys %ValidHostname) {
+			print "    $Message : $ValidHostname{$Message} Times(s)\n";
 		}
 	}
 	
